---
title: ""
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(stringr)
library(useful)
library(here)
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(sqldf)
library(matrixTests)
library(pryr)
library(VennDiagram)
library(ggbeeswarm)
library(ggridges)
library(ggpubr)
library(scales)
library(parallel)
library(statmod)
library(ggrepel)
library(rentrez)
library(biomaRt)

```

#Data loading and manipulation
```{r}

#Hugo symbol locus types, used to identify protein-coding genes
hugo_symbol_locus_types <- read_tsv(here::here("data","hugo_symbol_locus_types.txt"))
hugo_symbol_locus_types <- hugo_symbol_locus_types %>% rename(symbol = `Approved symbol`, name = `Approved name`, locus = `Locus type`)

#Cosmic fusion, cancer census gene, and kinase gene lists
cosmic_fusions <- read_tsv(here::here("data","cosmic_fusions.tsv"))
cosmic_fusion_genes_df_fusion_tibble <- read_tsv(here::here("data","cosmic_fusion_genes_df_fusion_tibble.tsv"))
cosmic_translocation_genes <- read_tsv(here::here("data","cosmic_translocation_genes.tsv"))
cosmic_census_genes <- read_tsv(here::here("data","cosmic_census_genes.tsv"))
oncogenes_tsps_kinases_drivers <- read_tsv(here::here("data","oncogenes_tsps_kinases_drivers.txt"))

#Nonessentials and common essentials used in Broad CRISPR screening
common_essentials <- read_csv(here::here("data","common_essentials.csv"))
common_essentials_gene_list <- str_split_fixed(common_essentials$gene, " ", 2)[,1]
nonessentials <- read_csv(here::here("data","nonessentials.csv"))
nonessentials_gene_list <- str_split_fixed(nonessentials$gene, " ", 2)[,1]

#CCLE fusion and annotation data
ccle_fusions <- read_tsv(here::here("data","CCLE_fusions.csv"))
ccle_annotations <- read_csv(here::here("data","sample_info.csv"))
ccle_fusions_annotated <- left_join(ccle_fusions, ccle_annotations, by = c("DepMap_ID" = "DepMap_ID"))
disease_annotations_by_cell_line <- ccle_fusions_annotated %>% distinct(DepMap_ID,CCLE_Name,lineage,lineage_subtype,primary_disease,Subtype)

ccle_fusions_annotated <- ccle_fusions_annotated %>% mutate(LeftGeneSymbol = str_split_fixed(LeftGene," ",2)[,1], LeftEnsemblID = gsub("[()]","",str_split_fixed(LeftGene," ",2)[,2]), RightGeneSymbol = str_split_fixed(RightGene," ",2)[,1], RightEnsemblID = gsub("[()]","",str_split_fixed(RightGene," ",2)[,2]), Fusion = str_c(LeftGeneSymbol,"-",RightGeneSymbol), LeftChrom = str_split_fixed(LeftBreakpoint,":",3)[,1], LeftChromPosition = str_split_fixed(LeftBreakpoint,":",3)[,2], LeftChromOrientation = str_split_fixed(LeftBreakpoint,":",3)[,3], RightChrom = str_split_fixed(RightBreakpoint,":",3)[,1], RightChromPosition = str_split_fixed(RightBreakpoint,":",3)[,2], RightChromOrientation = str_split_fixed(RightBreakpoint,":",3)[,3]) %>% select(-`#FusionName`,-LeftGene,-LeftBreakpoint,-RightGene,-RightBreakpoint)

ccle_fusions_annotated <- left_join(ccle_fusions_annotated, hugo_symbol_locus_types,by = c("LeftGeneSymbol" = "symbol")) %>% rename(LeftGeneName = name, LeftGeneType = locus)

ccle_fusions_annotated <- left_join(ccle_fusions_annotated ,hugo_symbol_locus_types,by = c("RightGeneSymbol" = "symbol")) %>% rename(RightGeneName = name, RightGeneType = locus)

ccle_fusions_annotated <- ccle_fusions_annotated %>% mutate(Cosmic = Fusion %in% cosmic_fusion_genes_df_fusion_tibble$value)

ccle_fusions_annotated <- ccle_fusions_annotated[,c("DepMap_ID","Fusion","Cosmic","LeftGeneSymbol","LeftEnsemblID","LeftGeneName","LeftGeneType","RightGeneSymbol","RightEnsemblID","RightGeneName","RightGeneType","LeftChrom","LeftChromPosition","LeftChromOrientation","RightChrom","RightChromPosition","RightChromOrientation","FFPM","JunctionReadCount","SpanningFragCount","SpliceType","LargeAnchorSupport","LeftBreakDinuc","LeftBreakEntropy","RightBreakDinuc","RightBreakEntropy","annots","CCLE_count","stripped_cell_line_name","CCLE_Name","Alias","COSMICID","lineage","lineage_subtype","lineage_sub_subtype","sex","source","Achilles_n_replicates","cell_line_NNMD","culture_type","culture_medium","cas9_activity","RRID","sample_collection_site","primary_or_metastasis","primary_disease","Subtype","age","Sanger_Model_ID")]

#CCLE RNA expression data for protein-coding genes
ccle_expression_protein_coding_log2tpm <- read_csv(here::here("data","CCLE_expression.csv"))
ccle_expression_protein_coding_log2tpm <- ccle_expression_protein_coding_log2tpm %>% rename(DepMap_ID = X1)
ccle_expression_protein_coding_log2tpm_colnames <- str_split_fixed(colnames(ccle_expression_protein_coding_log2tpm), " ",2)[,1]
colnames(ccle_expression_protein_coding_log2tpm) <- ccle_expression_protein_coding_log2tpm_colnames
ccle_expression_protein_coding_log2tpm_matrix <- ccle_expression_protein_coding_log2tpm %>% column_to_rownames("DepMap_ID") %>% t()
ccle_expression_protein_coding_log2tpm_matrix_df <- ccle_expression_protein_coding_log2tpm_matrix %>% as.data.frame() %>% rownames_to_column("gene")

#CCLE dependency data for protein-coding genes
ccle_dependency_probability <- read_csv(here::here("data","gene_dependency.csv"))
ccle_dependency_probability_colnames <- str_split_fixed(colnames(ccle_dependency_probability), " ",2)[,1]
colnames(ccle_dependency_probability) <- ccle_dependency_probability_colnames

#CCLE dependency sgRNA location data
ccle_dependency_guides <- read_csv(here::here("data","gene_guide_map.csv"))
ccle_dependency_guides <- ccle_dependency_guides %>% mutate(chromosome = str_split_fixed(genome_alignment,"_",3)[,1], base = str_split_fixed(genome_alignment,"_",3)[,2], direction = str_split_fixed(genome_alignment,"_",3)[,3], gene = str_split_fixed(gene, " ", 2)[,1])
ccle_dependency_guides <- ccle_dependency_guides %>% select(gene,chromosome,base,direction,n_alignments,sgrna)
ccle_dependency_probability_matrix <- ccle_dependency_probability %>% column_to_rownames("DepMap_ID") %>% t()
ccle_dependency_probability_matrix_df <- ccle_dependency_probability_matrix %>% as.data.frame() %>% rownames_to_column("gene")

#CCLE Copy number alterations - gene level
CCLE_gene_cn <- read_csv(here::here("data","CCLE_gene_cn.csv"))
CCLE_gene_cn <- CCLE_gene_cn %>% rename(DepMap_ID = X1)
CCLE_gene_cn <- CCLE_gene_cn %>% rename_all(funs(str_split_fixed(.," ",2)[,1]))

#CCLE Mutation data
CCLE_mutations <- read_tsv(here::here("data","CCLE_mutations.csv"))

#CCLE_translocations_SvABA_20181221 - structural variant calls on a subset of CCLE lines with WGS data. Of note, the chromosomal breakpoints are based on alignment to hg19

CCLE_translocations_SvABA_20181221 <- read_tsv(here::here("data","CCLE_translocations_SvABA_20181221.txt"))

CCLE_translocations_SvABA_20181221 <- CCLE_translocations_SvABA_20181221 %>% mutate(fusion = str_c(gene1,"-",gene2), fusion_inverse = str_c(gene2,"-",gene1))

#Biomart query for gene symbols and corresgene coordinates
ensembl <- useMart("ENSEMBL_MART_ENSEMBL")
datasets <- listDatasets(ensembl)
ensembl = useDataset("hsapiens_gene_ensembl",mart=ensembl) #hsapiens_gene_ensembl is version GRCh38.p13
filters = listFilters(ensembl)
attributes = listAttributes(ensembl)

gene_coordinate_table_hg38 <- getBM(attributes = c("hgnc_symbol","chromosome_name","start_position","end_position"), mart = ensembl)

gene_coordinate_table_hg38 <- gene_coordinate_table_hg38 %>% filter(!hgnc_symbol == (""))

gene_coordinate_table_hg38 <- gene_coordinate_table_hg38 %>% filter(chromosome_name %in% c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"))

gene_coordinate_table_hg38 <- gene_coordinate_table_hg38 %>% mutate(chromosome_name = str_c("chr",gene_coordinate_table_hg38$chromosome_name))

#TAD boundaries from a reference "normal" cell line
reference_tad_boundaries_hg38 <- read_tsv(here::here("data","ENCFF633ORE_endometrial_microvascular_endothelial_cells_hg38.bed"))

reference_tad_boundaries_hg38 <- reference_tad_boundaries_hg38 %>% mutate(chromosome = str_split_fixed(reference_tad_boundaries_hg38$TAD,":",2)[,1], coordinates = str_split_fixed(reference_tad_boundaries_hg38$TAD,":",2)[,2])

reference_tad_boundaries_hg38 <- reference_tad_boundaries_hg38 %>% mutate(start = str_split_fixed(reference_tad_boundaries_hg38$coordinates,"-",2)[,1], end = str_split_fixed(reference_tad_boundaries_hg38$coordinates,"-",2)[,2]) %>% dplyr::select(chromosome,start,end) %>% mutate(TAD_number = row_number())

reference_tad_boundaries_hg38$start <- as.integer(reference_tad_boundaries_hg38$start)

reference_tad_boundaries_hg38$end <- as.integer(reference_tad_boundaries_hg38$end)

#Gene_TAD_table - constructed by checking to see if start of gene lies within a particular TAD
Gene_TAD_table <- sqldf('SELECT gene_coordinate_table_hg38.*,reference_tad_boundaries_hg38.* FROM gene_coordinate_table_hg38 INNER JOIN reference_tad_boundaries_hg38 ON (gene_coordinate_table_hg38.start_position BETWEEN reference_tad_boundaries_hg38.start AND reference_tad_boundaries_hg38.end) AND (gene_coordinate_table_hg38.chromosome_name = reference_tad_boundaries_hg38.chromosome)')

Gene_TAD_table <- Gene_TAD_table %>% rename(TAD_chromosome = chromosome, TAD_start = start, TAD_end = end)

Gene_TAD_table <- Gene_TAD_table %>% rename(gene = hgnc_symbol, chromosome = chromosome_name, TAD_identifier = TAD_number)

#Gene_TAD_boundary_table - constructed for genes that fall into a TAD boundaries, to assign them to upstream and downstream TADs
genes_TAD_boundaries_coordinate_table_hg38 <- gene_coordinate_table_hg38 %>% filter(!hgnc_symbol %in% Gene_TAD_table$gene)

Gene_TAD_boundary_table_upstream_TAD <- sqldf('SELECT genes_TAD_boundaries_coordinate_table_hg38.*,reference_tad_boundaries_hg38.* FROM genes_TAD_boundaries_coordinate_table_hg38 INNER JOIN reference_tad_boundaries_hg38 ON (genes_TAD_boundaries_coordinate_table_hg38.start_position > reference_tad_boundaries_hg38.end) AND (genes_TAD_boundaries_coordinate_table_hg38.start_position - reference_tad_boundaries_hg38.end < 40000) AND (genes_TAD_boundaries_coordinate_table_hg38.chromosome_name = reference_tad_boundaries_hg38.chromosome)')

Gene_TAD_boundary_table_upstream_TAD <- Gene_TAD_boundary_table_upstream_TAD %>% dplyr::rename(gene = hgnc_symbol, upstream_TAD_identifier = TAD_number) %>% dplyr::select(gene, upstream_TAD_identifier)

Gene_TAD_boundary_table_downstream_TAD <- sqldf('SELECT genes_TAD_boundaries_coordinate_table_hg38.*,reference_tad_boundaries_hg38.* FROM genes_TAD_boundaries_coordinate_table_hg38 INNER JOIN reference_tad_boundaries_hg38 ON (genes_TAD_boundaries_coordinate_table_hg38.start_position < reference_tad_boundaries_hg38.start) AND (reference_tad_boundaries_hg38.start - genes_TAD_boundaries_coordinate_table_hg38.start_position < 40000) AND (genes_TAD_boundaries_coordinate_table_hg38.chromosome_name = reference_tad_boundaries_hg38.chromosome)')

Gene_TAD_boundary_table_downstream_TAD <- Gene_TAD_boundary_table_downstream_TAD %>% dplyr::rename(gene = hgnc_symbol, downstream_TAD_identifier = TAD_number) %>% dplyr::select(gene, downstream_TAD_identifier)

Gene_TAD_boundary_table <- full_join(Gene_TAD_boundary_table_upstream_TAD,Gene_TAD_boundary_table_downstream_TAD)

#Write output of Gene TAD table and Gene TAD boundary table
write_tsv(Gene_TAD_table,here::here("results/datatables","Gene_TAD_table.tsv"))

write_tsv(Gene_TAD_boundary_table,here::here("results/datatables","Gene_TAD_boundary_table.tsv"))

#TCGA fusion list called by RNAseq
tcga_fusion_list <- read_tsv(here::here("data","tcga_fusion_list.txt"))
tcga_fusion_list_symbols_updated <- tcga_fusion_list
tcga_fusion_list_symbols_updated$LeftGene <- recode(tcga_fusion_list_symbols_updated$LeftGene, "LHFP" = "LHFPL6", "MKL1" = "MRTFA")
tcga_fusion_list_symbols_updated$RightGene <- recode(tcga_fusion_list_symbols_updated$RightGene, "LHFP" = "LHFPL6", "MKL1" = "MRTFA")
tcga_study_abbreviations <- read_tsv(here::here("data","tcga_study_abbreviations.txt"))

#CTD2 drug screen data
ctd2_auc <- read_tsv(here::here("data/CTD2_drug_screen","v20.data.curves_post_qc.txt"))
ctd2_experiment_metadata <- read_tsv(here::here("data/CTD2_drug_screen","v20.meta.per_experiment.txt"))
ctd2_cell_metadata <- read_tsv(here::here("data/CTD2_drug_screen","v20.meta.per_cell_line.txt"))
ctd2_compound_metadata <- read_tsv(here::here("data/CTD2_drug_screen","v20.meta.per_compound.txt"))

#Lung cancer cells based organoid model CRISPR screen from "CRISPR screens in cancer spheroids identify 3D growth-specific vulnerabilities"
NCIH23_D19_3D_U1_counts <- read_csv(here::here("data/H23_GW_counts_organoid_study","D19_3D_U1_counts.csv"))
NCIH23_D19_3D_U2_counts <- read_csv(here::here("data/H23_GW_counts_organoid_study","D19_3D_U2_counts.csv"))
NCIH23_T0_counts <- read_csv(here::here("data/H23_GW_counts_organoid_study","T0_counts.csv"))
NCIH1975_D19_3D_U1_counts <- read_csv(here::here("data/H1975_GW_counts_organoid_study","H1975_3D_1_merged.csv"))
NCIH1975_D19_3D_U2_counts <- read_csv(here::here("data/H1975_GW_counts_organoid_study","H1975_3D_2_merged.csv"))
NCIH1975_T0_counts <- read_csv(here::here("data/H1975_GW_counts_organoid_study","H1975_T0_merged.csv"))
NCIH2009_D19_3D_U1_counts <- read_csv(here::here("data/H2009_GW_counts_organoid_study","H2009_3D_pooled_3.csv"))
NCIH2009_D19_3D_U2_counts <- read_csv(here::here("data/H2009_GW_counts_organoid_study","H2009_3D_pooled_4.csv"))
NCIH2009_T0_counts <- read_csv(here::here("data/H2009_GW_counts_organoid_study","H2009_T0_pooled.csv"))

#Fragile sites in the PCAWG dataset
fragile_sites_PCAWG_hg38 <- read_tsv(here::here("data","fragile_sites_PCAWG_hg38.bed"))

fragile_sites_PCAWG_hg38 <- fragile_sites_PCAWG_hg38 %>% mutate(chromosome = str_split_fixed(fragile_sites,"[:|-]",3)[,1], start = str_split_fixed(fragile_sites,"[:|-]",3)[,2], end = str_split_fixed(fragile_sites,"[:|-]",3)[,3])

#Venetoclax screening data and B-ALL cell lines
venetoclax_screen_all_cell_lines <- read_csv(here::here("data","venetoclax_screen_all_cell_lines.csv"))

B_ALL_cell_lines <- scan(here::here("data","B_ALL_cell_lines.txt"), what = "", sep = ",") %>% as_vector()
  
```

#Filtering fusion calls
```{r}

#List of fusions in CCLE filtered by FFPM > 0.1, GT/ AG breakpoints, JunctionReads > 5, Both genes either being protein coding or part of the IgH locus. Subsequent cleaning to uniformly label IgH locus.
ALL_fusions_protein_coding_filtered_list <- ccle_fusions_annotated %>% filter(FFPM > 0.1 & LeftBreakDinuc == "GT" & RightBreakDinuc == "AG" & JunctionReadCount > 5 & (LeftGeneType == "gene with protein product" | LeftGeneSymbol %in% str_subset(ccle_fusions_annotated$LeftGeneSymbol,"IGH")) & (RightGeneType == "gene with protein product" | RightGeneSymbol %in% str_subset(ccle_fusions_annotated$RightGeneSymbol,"IGH"))) %>% view()

ALL_fusions_protein_coding_filtered_list_IGH_fusions <- ALL_fusions_protein_coding_filtered_list %>% distinct(Fusion,LeftGeneType,RightGeneType) %>% filter(!LeftGeneType == "gene with protein product" | !RightGeneType == "gene with protein product" | LeftGeneType %>% is.na() | RightGeneType %>% is.na())

ALL_fusions_protein_coding_filtered_list <- ALL_fusions_protein_coding_filtered_list %>% filter(!Fusion %in% c("NBEA-IGHV1OR15-6","NSD2-IGHV5-78"))

ALL_fusions_protein_coding_filtered_list_without_IGH <- ALL_fusions_protein_coding_filtered_list %>% filter(!Fusion %in% ALL_fusions_protein_coding_filtered_list_IGH_fusions$Fusion)

ALL_fusions_protein_coding_filtered_list_with_IGH_left <- ALL_fusions_protein_coding_filtered_list %>% filter(Fusion %in% ALL_fusions_protein_coding_filtered_list_IGH_fusions$Fusion & LeftGeneSymbol %in% str_subset(ALL_fusions_protein_coding_filtered_list$LeftGeneSymbol,"IGH")) %>% mutate(LeftGeneSymbol = "IGH") %>% mutate(Fusion = str_c(LeftGeneSymbol,"-",RightGeneSymbol))

ALL_fusions_protein_coding_filtered_list_with_IGH_right <- ALL_fusions_protein_coding_filtered_list %>% filter(Fusion %in% ALL_fusions_protein_coding_filtered_list_IGH_fusions$Fusion & RightGeneSymbol %in% str_subset(ALL_fusions_protein_coding_filtered_list$RightGeneSymbol,"IGH")) %>% mutate(RightGeneSymbol = "IGH") %>% mutate(Fusion = str_c(LeftGeneSymbol,"-",RightGeneSymbol))

ALL_fusions_protein_coding_filtered_list_with_IGH <- full_join(ALL_fusions_protein_coding_filtered_list_with_IGH_left,ALL_fusions_protein_coding_filtered_list_with_IGH_right)

ALL_fusions_protein_coding_filtered_list_final <- full_join(ALL_fusions_protein_coding_filtered_list_without_IGH, ALL_fusions_protein_coding_filtered_list_with_IGH)

hyphenated_genes_to_filter_out <- ALL_fusions_protein_coding_filtered_list_final %>% filter(LeftGeneSymbol %in% str_subset(ALL_fusions_protein_coding_filtered_list_final$LeftGeneSymbol,"-") | RightGeneSymbol %in% str_subset(ALL_fusions_protein_coding_filtered_list_final$RightGeneSymbol,"-") ) %>% distinct(Fusion) %>% unlist() %>% unname()

ALL_fusions_protein_coding_filtered_list_final <- ALL_fusions_protein_coding_filtered_list_final %>% filter(!Fusion %in% hyphenated_genes_to_filter_out)

ALL_fusions_protein_coding_filtered_list_final_unique <- ALL_fusions_protein_coding_filtered_list_final %>% arrange(Fusion) %>% distinct(Fusion) %>% unlist() %>% unname()

#All fusions that have dependency data available
ALL_fusions_with_dependency_data <- tibble(Fusion=NA)
counter = 0

for(fusion in ALL_fusions_protein_coding_filtered_list_final_unique){
  
cell_lines_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

if(length(cell_lines_with_fusion) > 0){

ALL_fusions_with_dependency_data <- add_row(ALL_fusions_with_dependency_data, Fusion = fusion)

counter = counter + 1
print(counter)
  
}
}

#Note that while all of the fusions in this table appear in cell lines with dependency data, not all cell lines in this table have dependency. As an example, EWSR1-FLI1 appears in this list because there is at least one cell line with EWSR1-FLI1 with dependency data, but the other cell lines with EWSR1-FLI1 without dependency data also appear in this list. Therefore, the cell lines from this list must be intersected by the cell lines in the ccle_dependency_probability_matrix for an accurate view of high-confidence fusions only in cell lines with dependency data
ALL_fusions_protein_coding_filtered_list_final_with_dependency_data <- ALL_fusions_protein_coding_filtered_list_final %>% filter(Fusion %in% ALL_fusions_with_dependency_data)

ALL_fusions_protein_coding_filtered_list_final_with_dependency_data <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% mutate(in_cell_line_with_dependency_data = DepMap_ID %in% colnames(ccle_dependency_probability_matrix))

#Write output: Table of fusions with protein-coding/ IgH partners
write_tsv(ALL_fusions_protein_coding_filtered_list_final_with_dependency_data,here::here("results/datatables","ALL_fusions_protein_coding_filtered_list_final_with_dependency_data.tsv"))

```

#Genome-scale dependency and unbiased differential expression analysis of fusion-associated genes
```{r}

TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries <- tibble(Fusion=NA,Gene=NA,Cell_Count=NA)
TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries <- tibble(Fusion=NA,Gene=NA,Cell_Count=NA)

dependency_probability_fusion_with_vs_without_stats <- list()
dependency_probability_fusion_with_vs_without_stats_matrix <- list()
expression_fusion_with_vs_without_stats <- list()
expression_fusion_with_vs_without_stats_matrix <- list()

left_gene_TAD_neighbors <- list()
right_gene_TAD_neighbors <- list()
fusion_partners <- list()

left_gene_TAD_neighbors_expression_quartiles <- list()
right_gene_TAD_neighbors_expression_quartiles <- list()
fusion_partners_expression_quartiles <- list()

counter = 0

for(fusion in ALL_fusions_with_dependency_data){
  
#Cell lines with dependency data that have fusion of interest and those that don't
cell_lines_with_fusion_dependency <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

cell_lines_without_fusion_dependency <- ccle_dependency_probability %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_dependency)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#Cell lines with expression data that have fusion of interest and those that don't
cell_lines_with_fusion_expression <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_expression_protein_coding_log2tpm_matrix))

cell_lines_without_fusion_expression <- ccle_expression_protein_coding_log2tpm %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_expression)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#Fusion partners
left_gene_fusion <- str_split_fixed(fusion,"-",2)[,1]
right_gene_fusion <- str_split_fixed(fusion,"-",2)[,2]
fusion_partners[[fusion]] <- c(left_gene_fusion,right_gene_fusion)

#If there is at least one cell line that has dependency data, then proceed
if(length(cell_lines_with_fusion_dependency) > 0 & length(cell_lines_without_fusion_dependency) > 0){
  
#Create expression matrices for cell lines that have the fusion and cell lines that don't
ccle_expression_matrix_with_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_with_fusion_expression)

ccle_expression_matrix_without_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_without_fusion_expression)

#Compare expression between the two groups with two sample t-test with equal variance and correct p-values using Benjamini-Hochberg procedure for multiple hypothesis testing. Also categorize by quartile of differential expression. Then create an abstracted binary matrix
expression_fusion_with_vs_without_stats[[fusion]] <- row_t_equalvar(ccle_expression_matrix_with_fusion, ccle_expression_matrix_without_fusion) %>% dplyr::mutate(BH_QValue = p.adjust(pvalue, "BH", length(ccle_expression_matrix_with_fusion[,1]))) %>% dplyr::mutate(negative_log10_BH_QValue = -log10(BH_QValue), gene = ccle_expression_protein_coding_log2tpm_matrix_df$gene) %>% dplyr::rename(log2_fold_change = `mean.diff`, expression_with_fusion = `mean.x`, expression_without_fusion = `mean.y`) %>% dplyr::select(gene, log2_fold_change, expression_with_fusion, expression_without_fusion, negative_log10_BH_QValue) %>% mutate(log2_fold_change_quartile = ntile(log2_fold_change,4))

expression_fusion_with_vs_without_stats_matrix[[fusion]] <- expression_fusion_with_vs_without_stats[[fusion]] %>% mutate(overexpressed = ifelse(log2_fold_change > 1,1,0), significant = ifelse(negative_log10_BH_QValue > -log10(.05),1,0), overexpressed_and_significant = ifelse(overexpressed == 1 & significant == 1, 1, 0)) %>% select(overexpressed_and_significant,log2_fold_change_quartile)

#Create dependency matrices for cell lines that have the fusion and cell lines that don't  
ccle_dependency_probability_matrix_with_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_with_fusion_dependency)

ccle_dependency_probability_matrix_without_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_without_fusion_dependency)

#Compare dependency between the two groups with two sample t-test with equal variance and correct p-values using Benjamini-Hochberg procedure for multiple hypothesis testing. Also categorize by quartile of differential expression (joined from above). Then create an abstracted binary matrix
dependency_probability_fusion_with_vs_without_stats[[fusion]] <- row_t_equalvar(ccle_dependency_probability_matrix_with_fusion, ccle_dependency_probability_matrix_without_fusion) %>% dplyr::mutate(BH_QValue = p.adjust(pvalue, "BH", length(ccle_dependency_probability_matrix_with_fusion[,1]))) %>% dplyr::mutate(negative_log10_BH_QValue = -log10(BH_QValue), gene = ccle_dependency_probability_matrix_df$gene) %>% dplyr::rename(dependency_probability_difference = `mean.diff`, dependency_probability_with_fusion = `mean.x`, dependency_probability_without_fusion = `mean.y`) %>% dplyr::select(gene, dependency_probability_difference, dependency_probability_with_fusion, dependency_probability_without_fusion, negative_log10_BH_QValue)

dependency_probability_fusion_with_vs_without_stats[[fusion]] <- merge(dependency_probability_fusion_with_vs_without_stats[[fusion]],expression_fusion_with_vs_without_stats[[fusion]][,c("gene","log2_fold_change_quartile")],by = c("gene","gene"),all.x = TRUE)

dependency_probability_fusion_with_vs_without_stats_matrix[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% mutate(dependency = ifelse(dependency_probability_difference > 0.5,1,0), significant = ifelse(negative_log10_BH_QValue > -log10(.05),1,0), dependency_and_significant = ifelse(dependency == 1 & significant == 1, 1, 0)) %>% select(dependency_and_significant,log2_fold_change_quartile)

#If the partner gene falls into a TAD then proceed to identify the other genes that fall into that TAD that are not the fusion partners themselves
if(left_gene_fusion %in% Gene_TAD_table$gene){

left_gene_TAD <- Gene_TAD_table %>% filter(gene == left_gene_fusion) %>% dplyr::select(TAD_identifier) %>% unlist() %>% unname()
left_gene_TAD_neighbors[[fusion]] <- Gene_TAD_table %>% filter(TAD_identifier == left_gene_TAD) %>% dplyr::select(gene) %>% unlist() %>% unname()
left_gene_TAD_neighbors[[fusion]] <- left_gene_TAD_neighbors[[fusion]][!left_gene_TAD_neighbors[[fusion]] %in% c(left_gene_fusion,right_gene_fusion)]

#If the partner gene falls into a TAD boundary (and not a single TAD) then look to see if there is an upstream or downstream TAD that is within 40Kb of the gene. If so, identify the genes that fall into the upstream and/or downstream TADs (should not include the fusion partners themselves)
} else if(left_gene_fusion %in% Gene_TAD_boundary_table$gene){
left_gene_upstream_TAD <- Gene_TAD_boundary_table %>% filter(gene == left_gene_fusion) %>% dplyr::select(upstream_TAD_identifier) %>% unlist() %>% unname()

left_gene_downstream_TAD <- Gene_TAD_boundary_table %>% filter(gene == left_gene_fusion) %>% dplyr::select(downstream_TAD_identifier) %>% unlist() %>% unname()

left_gene_TADs <- c(left_gene_upstream_TAD,left_gene_downstream_TAD)

left_gene_TADs <- left_gene_TADs[!is.na(left_gene_TADs)]

left_gene_TAD_neighbors[[fusion]] <- Gene_TAD_table %>% filter(TAD_identifier %in% left_gene_TADs) %>% dplyr::select(gene) %>% unlist() %>% unname()

#If the partner gene has no plausibly associated TADs, say there are no genes that are close to it
} else{
left_gene_TAD_neighbors[[fusion]] <- c()  
}

#If the partner gene falls into a TAD then proceed to identify the other genes that fall into that TAD that are not the fusion partners themselves
if(right_gene_fusion %in% Gene_TAD_table$gene){
  
right_gene_TAD <- Gene_TAD_table %>% filter(gene == right_gene_fusion) %>% dplyr::select(TAD_identifier) %>% unlist() %>% unname()
right_gene_TAD_neighbors[[fusion]] <- Gene_TAD_table %>% filter(TAD_identifier == right_gene_TAD) %>% dplyr::select(gene) %>% unlist() %>% unname()
right_gene_TAD_neighbors[[fusion]] <- right_gene_TAD_neighbors[[fusion]][!right_gene_TAD_neighbors[[fusion]] %in% c(left_gene_fusion,right_gene_fusion)]

#If the partner gene falls into a TAD boundary (and not a single TAD) then look to see if there is an upstream or downstream TAD that is within 40Kb of the gene. If so, identify the genes that fall into the upstream and/or downstream TADs (should not include the fusion partners themselves)
} else if(right_gene_fusion %in% Gene_TAD_boundary_table$gene){
right_gene_upstream_TAD <- Gene_TAD_boundary_table %>% filter(gene == right_gene_fusion) %>% dplyr::select(upstream_TAD_identifier) %>% unlist() %>% unname()

right_gene_downstream_TAD <- Gene_TAD_boundary_table %>% filter(gene == right_gene_fusion) %>% dplyr::select(downstream_TAD_identifier) %>% unlist() %>% unname()

right_gene_TADs <- c(right_gene_upstream_TAD,right_gene_downstream_TAD)

right_gene_TADs <- right_gene_TADs[!is.na(right_gene_TADs)]

right_gene_TAD_neighbors[[fusion]] <- Gene_TAD_table %>% filter(TAD_identifier %in% right_gene_TADs) %>% dplyr::select(gene) %>% unlist() %>% unname()

#If the partner gene has no plausibly associated TADs, say there are no genes that are close to it
} else{
right_gene_TAD_neighbors[[fusion]] <- c()  
}

#Summary stats dependency tables for the genes in close proximity to the fusion partners
dependency_probability_fusion_with_vs_without_stats_left_gene_TAD_neighbors <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% dplyr::filter(gene %in% c(left_gene_TAD_neighbors[[fusion]], left_gene_fusion))

dependency_probability_fusion_with_vs_without_stats_right_gene_TAD_neighbors <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% dplyr::filter(gene %in% c(right_gene_TAD_neighbors[[fusion]],right_gene_fusion))

#Summary stats expression tables for the genes in close proximity to the fusion partners
expression_fusion_with_vs_without_stats_left_gene_TAD_neighbors <- expression_fusion_with_vs_without_stats[[fusion]] %>% dplyr::filter(gene %in% c(left_gene_TAD_neighbors[[fusion]], left_gene_fusion))

expression_fusion_with_vs_without_stats_right_gene_TAD_neighbors <- expression_fusion_with_vs_without_stats[[fusion]] %>% dplyr::filter(gene %in% c(right_gene_TAD_neighbors[[fusion]],right_gene_fusion))

#For left partner or neighbors, if the gene has dependency data and is a significant dependency, then add the fusion, gene, and number of cells to the total signficant fusion-associated dependency table. Same if overexpression is significant
for(neighbor in c(left_gene_TAD_neighbors[[fusion]], left_gene_fusion)){

if(neighbor %in% ccle_dependency_probability_matrix_df$gene){
  
if(!subset(dependency_probability_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = dependency_probability_difference) %>% unlist() %>% unname() == "NaN"){
  
if(subset(dependency_probability_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = dependency_probability_difference) %>% unlist() %>% unname() >= 0.5 & subset(dependency_probability_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = negative_log10_BH_QValue) %>% unlist() %>% unname() >= -log10(.05)){
  
TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries <- add_row(TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries, Fusion = fusion, Gene = neighbor, Cell_Count = length(cell_lines_with_fusion_dependency))

}
}
}
  
if(neighbor %in% ccle_expression_protein_coding_log2tpm_matrix_df$gene){
  
if(!subset(expression_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = log2_fold_change) %>% unlist() %>% unname() == "NaN"){
  
if(subset(expression_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = log2_fold_change) %>% unlist() %>% unname() >= 1 & subset(expression_fusion_with_vs_without_stats_left_gene_TAD_neighbors, subset = gene == neighbor, select = negative_log10_BH_QValue) %>% unlist() %>% unname() >= -log10(.05)){
  
TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries <- add_row(TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries, Fusion = fusion, Gene = neighbor, Cell_Count = length(cell_lines_with_fusion_expression))
  
}
}
}
  
}

#For right partner or neighbors, if the gene has dependency data and is a significant dependency, then add the fusion, gene, and number of cells to the total signficant fusion-associated dependency table. Same if overexpression is significant  
for(neighbor in c(right_gene_TAD_neighbors[[fusion]],right_gene_fusion)){

if(neighbor %in% ccle_dependency_probability_matrix_df$gene){
  
if(!subset(dependency_probability_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = dependency_probability_difference) %>% unlist() %>% unname() == "NaN"){
  
if(subset(dependency_probability_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = dependency_probability_difference) %>% unlist() %>% unname() >= 0.5 & subset(dependency_probability_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = negative_log10_BH_QValue) %>% unlist() %>% unname() >= -log10(.05)){
  
TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries <- add_row(TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries, Fusion = fusion, Gene = neighbor, Cell_Count = length(cell_lines_with_fusion_dependency))
  
}
}
}
  
if(neighbor %in% ccle_expression_protein_coding_log2tpm_matrix_df$gene){
  
if(!subset(expression_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = log2_fold_change) %>% unlist() %>% unname() == "NaN"){
  
if(subset(expression_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = log2_fold_change) %>% unlist() %>% unname() >= 1 & subset(expression_fusion_with_vs_without_stats_right_gene_TAD_neighbors, subset = gene == neighbor, select = negative_log10_BH_QValue) %>% unlist() %>% unname() >= -log10(.05)){
  
TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries <- add_row(TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries, Fusion = fusion, Gene = neighbor, Cell_Count = length(cell_lines_with_fusion_expression))

}
}
}  
  
}

#Make vectors of the expression quartiles of left partner TAD associated genes, right partner TAD associated genes, and both partners. Note that this inherently filters these lists for 1. genes are protein-coding and 2. genes that have associated dependency data (almost one to one, but not exactly)
left_gene_TAD_neighbors_expression_quartiles[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% left_gene_TAD_neighbors[[fusion]]) %>% select(log2_fold_change_quartile) %>% unlist() %>% unname()

right_gene_TAD_neighbors_expression_quartiles[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% right_gene_TAD_neighbors[[fusion]]) %>% select(log2_fold_change_quartile) %>% unlist() %>% unname() 

fusion_partners_expression_quartiles[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% fusion_partners[[fusion]]) %>% select(log2_fold_change_quartile) %>% unlist() %>% unname() 

}

counter = counter + 1
print(counter)

}

TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries %>% remove_missing()
TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries %>% mutate(is_partner = Gene == str_split_fixed(Fusion,"-",2)[,1] | Gene == str_split_fixed(Fusion,"-",2)[,2])

TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries <- TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries %>% remove_missing()
TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries <- TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries %>% mutate(is_partner = Gene == str_split_fixed(Fusion,"-",2)[,1] | Gene == str_split_fixed(Fusion,"-",2)[,2])

#Create list by fusion of all unique coding TAD neighbors with dependency data available (which are the lists that should be used for permutation). Doing this to avoid "double counting" for permutations later on
all_unique_coding_TAD_neighbors_with_dependency_data <- list()
all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles <- list()
counter = 0

for(fusion in ALL_fusions_with_dependency_data){

all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% c(left_gene_TAD_neighbors[[fusion]], right_gene_TAD_neighbors[[fusion]])) %>% select(gene) %>% unlist() %>% unname() 

counter = counter + 1
print(counter)
  
}

#Lists of corresponding expression quartiles for all unique coding TAD neighbors with dependency data available
for(fusion in ALL_fusions_with_dependency_data){

all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]] <- dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]]) %>% select(log2_fold_change_quartile) %>% unlist() %>% unname()     

counter = counter + 1
print(counter)
  
}

```

#Count of fusion-associated dependencies from genome-scale screen
```{r}

TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_marked <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries %>% mutate(Fusion_Inverse = str_c(str_split_fixed(Fusion,"-",2)[,2],"-",str_split_fixed(Fusion,"-",2)[,1])) %>% mutate(Duplicate = Fusion_Inverse %in% TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries$Fusion)

TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_marked %>% filter(!Fusion %in% c("ABL1-BCR", "C2orf69-ROCK2", "HNRNPC-RAD51B", "RARA-PML"))

#Distinct Fusion-Dependency Pairings
distinct_fusion_dependency_pairings <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed %>% distinct(Fusion,Gene,is_partner)

count_of_partner_fusion_dependency_pairings <- distinct_fusion_dependency_pairings %>% filter(is_partner == TRUE) %>% summarise(n_distinct(Fusion,Gene)) %>% unlist() %>% unname()

count_of_actual_fusion_dependency_pairings <- distinct_fusion_dependency_pairings %>% filter(is_partner == FALSE) %>% summarise(n_distinct(Fusion,Gene)) %>% unlist() %>% unname()

#List of dependencies identified and the number of fusions they are associated with
distinct_fusion_dependency_pairings_count_of_fusions <- distinct_fusion_dependency_pairings %>% group_by(Gene) %>% dplyr::summarise(count_of_fusions = n_distinct(Fusion)) %>% arrange(desc(count_of_fusions))

#For each unique fusion-dependency pairing, list all the cell lines in which it is seen
cell_lines_associated_with_each_gene_and_fusion <- tibble(gene = NA, fusion = NA, cell_lines = NA, CCLE_names = NA)

for(gene in unique(distinct_fusion_dependency_pairings$Gene)){
  
fusions_with_gene_as_dependency <- distinct_fusion_dependency_pairings %>% filter(Gene == gene) %>% distinct(Fusion) %>% unlist() %>% unname()

for(fusion in fusions_with_gene_as_dependency){

cell_lines_associated_with_each_gene_and_fusion <- add_row(cell_lines_associated_with_each_gene_and_fusion, gene = gene, fusion = fusion, cell_lines = str_c(ALL_fusions_protein_coding_filtered_list_final %>% filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname(), collapse = ", "), CCLE_names = str_c(ALL_fusions_protein_coding_filtered_list_final %>% filter(Fusion == fusion) %>% distinct(CCLE_Name) %>% unlist() %>% unname(), collapse = ", "))
  
}

}

cell_lines_associated_with_each_gene_and_fusion <- cell_lines_associated_with_each_gene_and_fusion %>% remove_missing()

cell_lines_associated_with_each_gene_and_fusion <- cell_lines_associated_with_each_gene_and_fusion %>% mutate(is_partner = gene == str_split_fixed(fusion,"-",2)[,1] | gene == str_split_fixed(fusion,"-",2)[,2])

cell_lines_associated_with_each_gene_and_fusion <- cell_lines_associated_with_each_gene_and_fusion %>% mutate(cell_line_count = str_count(cell_lines,"ACH"))

cell_lines_associated_with_each_gene_and_fusion <- cell_lines_associated_with_each_gene_and_fusion %>% mutate(left_fusion_partner = str_split_fixed(fusion,"-",2)[,1], right_fusion_partner = str_split_fixed(fusion,"-",2)[,2])

cell_lines_associated_with_each_gene_and_fusion <- cell_lines_associated_with_each_gene_and_fusion %>% select(fusion,left_fusion_partner,right_fusion_partner,gene,is_partner,cell_lines,CCLE_names,cell_line_count)

#Develop a conservative count for the number of fusion-dependency pairings by removing instances in which the same dependency is associated with multiple fusions in the same cell line (to account for complex rearrangements). This is used as the comparator for the number of fusion-dependency pairings expected by chance in subsequent permutation analysis.
cell_lines_associated_with_each_gene_and_fusion_with_dependency_counts <- tibble(fusion = NA, cell_lines_with_dependency_data = NA, CCLE_names_with_dependency_data = NA, cell_line_count_with_dependency_data = NA)

for(fusion in cell_lines_associated_with_each_gene_and_fusion$fusion){
  
cell_lines_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

CCLE_names_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% filter(DepMap_ID %in% cell_lines_with_fusion) %>% distinct(CCLE_Name) %>% unlist() %>% unname()

cell_lines_associated_with_each_gene_and_fusion_with_dependency_counts <- add_row(cell_lines_associated_with_each_gene_and_fusion_with_dependency_counts, fusion = fusion, cell_lines_with_dependency_data = str_c(cell_lines_with_fusion, collapse = ", "), CCLE_names_with_dependency_data = str_c(CCLE_names_with_fusion, collapse = ", "), cell_line_count_with_dependency_data = str_count(cell_lines_with_dependency_data,"ACH"))
  
}

cell_lines_associated_with_each_gene_and_fusion_with_dependency_counts <- cell_lines_associated_with_each_gene_and_fusion_with_dependency_counts %>% remove_missing()

distinct_fusion_dependency_cell_line_pairings <- cell_lines_associated_with_each_gene_and_fusion %>% distinct(gene,cell_lines,is_partner) %>% arrange(desc(is_partner)) %>% arrange(gene,cell_lines) 

distinct_fusion_dependency_cell_line_pairings <- distinct_fusion_dependency_cell_line_pairings %>% mutate(duplicate = duplicated(str_c(gene,cell_lines)))

distinct_fusion_dependency_cell_line_pairings_duplicates_removed <- distinct_fusion_dependency_cell_line_pairings %>% filter(duplicate == FALSE)

count_of_partner_fusion_dependency_pairings_conservative <- distinct_fusion_dependency_cell_line_pairings_duplicates_removed %>% filter(is_partner == TRUE) %>% summarise(n_distinct(gene,cell_lines)) %>% unlist() %>% unname()

count_of_other_TAD_fusion_dependency_pairings_conservative <- distinct_fusion_dependency_cell_line_pairings_duplicates_removed %>% filter(is_partner == FALSE) %>% summarise(n_distinct(gene,cell_lines)) %>% unlist() %>% unname()

count_of_actual_fusion_dependency_pairings_conservative <- tibble(partner_or_other_TAD = c("partner","other_TAD"), count = c(count_of_partner_fusion_dependency_pairings_conservative, count_of_other_TAD_fusion_dependency_pairings_conservative))

#Count of all dependencies fusion context
count_of_all_dependencies_for_fusion_context <- tibble(fusion = NA, count_of_all_dependencies = NA)

count = 0

for(fusion in ALL_fusions_with_dependency_data){
  
count_of_all_dependencies_for_fusion_context <- add_row(count_of_all_dependencies_for_fusion_context,fusion = fusion, count_of_all_dependencies = dependency_probability_fusion_with_vs_without_stats_matrix[[fusion]] %>% summarise(sum(dependency_and_significant, na.rm = TRUE)) %>% unlist() %>% unname())

count = count + 1
print(count)

}

count_of_all_dependencies_for_fusion_context <- count_of_all_dependencies_for_fusion_context %>% remove_missing()

#Write output of fusion-dependency pairings
write_tsv(TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed, here::here("results/datatables","TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed.tsv"))

write_tsv(distinct_fusion_dependency_pairings, here::here("results/datatables","distinct_fusion_dependency_pairings.tsv"))

```

#Count of fusion-associated overexpressed genes from genome-scale screen
```{r}

TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_marked <- TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries %>% mutate(Fusion_Inverse = str_c(str_split_fixed(Fusion,"-",2)[,2],"-",str_split_fixed(Fusion,"-",2)[,1])) %>% mutate(Duplicate = Fusion_Inverse %in% TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries$Fusion)

TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed <- TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_marked %>% filter(!Fusion %in% c("ABL1-BCR", "NSD2-IGH"))

#Write output of fusion-overexpressed gene pairings
write_tsv(TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed, here::here("results/datatables","TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed.tsv"))

```

#Permuting gene labels to determine total number of fusion-associated dependencies expected by chance from genome-scale screen
```{r}

iteration_one_thousand <- c(1:1000)

multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles <- c()
multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles <- c()

iteration_counter = 1

for(i in iteration_one_thousand){

cumulative_partner_dependencies <- c()    
cumulative_other_TAD_dependencies <- c()

fusion_counter = 0

for(fusion in ALL_fusions_with_dependency_data){

#Binary matrix of significant dependencies (0 = no, 1 = yes) and associated expression quartile
dependency_and_significant <- dependency_probability_fusion_with_vs_without_stats_matrix[[fusion]][,c("dependency_and_significant","log2_fold_change_quartile")] 
dependency_and_significant <- dependency_and_significant %>% filter(!dependency_and_significant %>% is.na())

#Binary matrices of significant dependencies filtered for expression quartile
dependency_and_significant_first_quartile_expression <- dependency_and_significant %>% filter(log2_fold_change_quartile == 1) %>% select(dependency_and_significant) %>% unlist() %>% unname()
dependency_and_significant_second_quartile_expression <- dependency_and_significant %>% filter(log2_fold_change_quartile == 2) %>% select(dependency_and_significant) %>% unlist() %>% unname()
dependency_and_significant_third_quartile_expression <- dependency_and_significant %>% filter(log2_fold_change_quartile == 3) %>% select(dependency_and_significant) %>% unlist() %>% unname()
dependency_and_significant_fourth_quartile_expression <- dependency_and_significant %>% filter(log2_fold_change_quartile == 4) %>% select(dependency_and_significant) %>% unlist() %>% unname()

#Individually sample the number of partners with dependency data from each of the binary matrices, and sum these samples together. This is the equivalent of permuting all the genes.
sample_partner_dependency <- sum(dependency_and_significant_first_quartile_expression %>% sample(length(fusion_partners_expression_quartiles[[fusion]][fusion_partners_expression_quartiles[[fusion]] == 1])), dependency_and_significant_second_quartile_expression %>% sample(length(fusion_partners_expression_quartiles[[fusion]][fusion_partners_expression_quartiles[[fusion]] == 2])), dependency_and_significant_third_quartile_expression %>% sample(length(fusion_partners_expression_quartiles[[fusion]][fusion_partners_expression_quartiles[[fusion]] == 3])), dependency_and_significant_fourth_quartile_expression %>% sample(length(fusion_partners_expression_quartiles[[fusion]][fusion_partners_expression_quartiles[[fusion]] == 4])))

cumulative_partner_dependencies <- c(cumulative_partner_dependencies, sample_partner_dependency)

#Individually sample the number of unique coding TAD neighbors with dependency data from each of the binary matrices, and sum these samples together. This is the equivalent of permuting all the genes.
sample_other_TAD_dependency <- sum(dependency_and_significant_first_quartile_expression %>% sample(length(all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]][all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]] == 1])), dependency_and_significant_second_quartile_expression %>% sample(length(all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]][all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]] == 2])), dependency_and_significant_third_quartile_expression %>% sample(length(all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]][all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]] == 3])), dependency_and_significant_fourth_quartile_expression %>% sample(length(all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]][all_unique_coding_TAD_neighbors_with_dependency_data_expression_quartiles[[fusion]] == 4])))

cumulative_other_TAD_dependencies <- c(cumulative_other_TAD_dependencies, sample_other_TAD_dependency)

fusion_counter = fusion_counter + 1
print(fusion_counter)

}

multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles <- c(multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles, cumulative_partner_dependencies %>% sum())

multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles <- c(multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles, cumulative_other_TAD_dependencies %>% sum())

iteration_counter = iteration_counter + 1
print(str_c("Iteration = ", iteration_counter))

}

#Write output of fusion-dependency pairings expected from gene label permutation
write_tsv(as_tibble(multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles),here::here("results/datatables","multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles.tsv"))

write_tsv(as_tibble(multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles),here::here("results/datatables","multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles.tsv"))

```

#Permuting fusion labels to determine total number of fusion-associated dependencies expected by chance from genome-scale screen
```{r}
#Creating a list of dependency probability dataframes that can be randomly shuffled

dependency_probability_fusion_with_vs_without_stats_random <- dependency_probability_fusion_with_vs_without_stats

counter = 0

for(fusion in dependency_probability_fusion_with_vs_without_stats %>% names()){

dependency_probability_fusion_with_vs_without_stats_random[[fusion]] <- dependency_probability_fusion_with_vs_without_stats_random[[fusion]] %>% mutate(significant_dependency = ifelse(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05),1,0))

counter = counter + 1
print(counter)
  
}

#Defining functions to use in apply functions to determine number of partner or other TAD dependencies

count_of_partner_dependencies_function <- function(x){
dependency_probability_fusion_with_vs_without_stats_random[[x]] %>% filter(gene %in% fusion_partners[[x]]) %>% select(significant_dependency) %>% unlist() %>% unname() %>% sum()
}

count_of_other_TAD_dependencies_function <- function(x){
dependency_probability_fusion_with_vs_without_stats_random[[x]] %>% filter(gene %in% c(left_gene_TAD_neighbors[[x]], right_gene_TAD_neighbors[[x]])) %>% select(significant_dependency) %>% unlist() %>% unname() %>% sum()
}

#Lists to store information from permutations
count_of_partner_dependencies_phenotypic_permutation <- c()
list_of_fusions_with_partner_dependencies_phenotypic_permutation <- tibble(fusions=NA)
count_of_other_TAD_dependencies_phenotypic_permutation <- c()
list_of_fusions_with_other_TAD_dependencies_phenotypic_permutation <- tibble(fusions=NA)

counter = 0
num_cores <- detectCores() - 1

#Start of iterative for loop
for(i in iteration_one_thousand){
  
#Shuffle fusions assigned to different dependencies
ALL_fusions_with_dependency_data_random <- sample(ALL_fusions_with_dependency_data, ALL_fusions_with_dependency_data %>% length())
names(dependency_probability_fusion_with_vs_without_stats_random) <- ALL_fusions_with_dependency_data_random

#Create a cluster to parallelize
cl <- makeCluster(num_cores, type="FORK")

count_of_partner_dependencies <- parSapply(cl,ALL_fusions_with_dependency_data_random,count_of_partner_dependencies_function)
count_of_partner_dependencies <- count_of_partner_dependencies %>% replace_na(0)
count_of_partner_dependencies_phenotypic_permutation <- c(count_of_partner_dependencies_phenotypic_permutation, count_of_partner_dependencies[count_of_partner_dependencies>0] %>% sum())
list_of_fusions_with_partner_dependencies_phenotypic_permutation <- add_row(list_of_fusions_with_partner_dependencies_phenotypic_permutation, fusions = count_of_partner_dependencies[count_of_partner_dependencies>0] %>% names())

count_of_other_TAD_dependencies <- parSapply(cl,ALL_fusions_with_dependency_data_random,count_of_other_TAD_dependencies_function)
count_of_other_TAD_dependencies <- count_of_other_TAD_dependencies %>% replace_na(0)
count_of_other_TAD_dependencies_phenotypic_permutation <- c(count_of_other_TAD_dependencies_phenotypic_permutation, count_of_other_TAD_dependencies[count_of_other_TAD_dependencies>0] %>% sum())
list_of_fusions_with_other_TAD_dependencies_phenotypic_permutation <- add_row(list_of_fusions_with_other_TAD_dependencies_phenotypic_permutation, fusions = count_of_other_TAD_dependencies[count_of_other_TAD_dependencies>0] %>% names())

#Stop Cluster
stopCluster(cl)

counter = counter + 1
print(counter)

}

#Write output of fusion-dependency pairings expected from fusion label permutation
write_tsv(as_tibble(count_of_partner_dependencies_phenotypic_permutation),here::here("results/datatables","count_of_partner_dependencies_phenotypic_permutation.tsv"))

write_tsv(as_tibble(count_of_other_TAD_dependencies_phenotypic_permutation),here::here("results/datatables","count_of_other_TAD_dependencies_phenotypic_permutation.tsv"))

```

#Fusion-associated dependency t-statistic analysis: 1) Generate table of t-statistics for dependency (and expression) for each gene for each fusion
```{r}

dependency_probability_fusion_with_vs_without_TSTATS <- list()
expression_fusion_with_vs_without_TSTATS <- list()

counter = 0

for(fusion in ALL_fusions_with_dependency_data){
  
#Cell lines with dependency data that have fusion of interest and those that don't
cell_lines_with_fusion_dependency <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

cell_lines_without_fusion_dependency <- ccle_dependency_probability %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_dependency)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#Cell lines with expression data that have fusion of interest and those that don't
cell_lines_with_fusion_expression <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_expression_protein_coding_log2tpm_matrix))

cell_lines_without_fusion_expression <- ccle_expression_protein_coding_log2tpm %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_expression)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#If there is at least one cell line that has dependency data, then proceed
if(length(cell_lines_with_fusion_dependency) > 0 & length(cell_lines_without_fusion_dependency) > 0){
  
#Create expression matrices for cell lines that have the fusion and cell lines that don't
ccle_expression_matrix_with_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_with_fusion_expression)

ccle_expression_matrix_without_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_without_fusion_expression)

#Compare expression between the two groups with two sample t-test with equal variance
expression_fusion_with_vs_without_TSTATS[[fusion]] <- row_t_equalvar(ccle_expression_matrix_with_fusion, ccle_expression_matrix_without_fusion) %>% dplyr::mutate(gene = ccle_expression_protein_coding_log2tpm_matrix_df$gene) %>% dplyr::rename(log2_fold_change = `mean.diff`, expression_with_fusion = `mean.x`, expression_without_fusion = `mean.y`) %>% dplyr::select(gene, log2_fold_change, expression_with_fusion, expression_without_fusion, pvalue, statistic)

#Create dependency matrices for cell lines that have the fusion and cell lines that don't  
ccle_dependency_probability_matrix_with_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_with_fusion_dependency)

ccle_dependency_probability_matrix_without_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_without_fusion_dependency)

#Compare dependency between the two groups with two sample t-test with equal variance
dependency_probability_fusion_with_vs_without_TSTATS[[fusion]] <- row_t_equalvar(ccle_dependency_probability_matrix_with_fusion, ccle_dependency_probability_matrix_without_fusion) %>% dplyr::mutate(gene = ccle_dependency_probability_matrix_df$gene) %>% dplyr::rename(dependency_probability_difference = `mean.diff`, dependency_probability_with_fusion = `mean.x`, dependency_probability_without_fusion = `mean.y`) %>% dplyr::select(gene, dependency_probability_difference, dependency_probability_with_fusion, dependency_probability_without_fusion, pvalue, statistic)

}

counter = counter + 1
print(counter)

}

```

#Fusion-associated dependency t-statistic analysis: 2) Generate null tables of t-statistics for dependency based on the number of cell lines that have a fusion of interest (ranging from 1-11)
```{r}

cell_permutation_matrices_TSTATS_1000 <- list()

for(number_of_cells in c(1:11)){

cell_permutation_matrices_TSTATS_1000[[number_of_cells]] <- matrix(nrow = length(c(1:1000)), ncol = length(rownames(ccle_dependency_probability_matrix)))
rownames(cell_permutation_matrices_TSTATS_1000[[number_of_cells]]) <- c(1:1000)
colnames(cell_permutation_matrices_TSTATS_1000[[number_of_cells]]) <- rownames(ccle_dependency_probability_matrix)

count <- 0

for(iteration in c(1:1000)){

#Cell lines with dependency data that have fusion of interest and those that don't
cell_lines_with_fusion_dependency <- sample(colnames(ccle_dependency_probability_matrix), number_of_cells)

cell_lines_without_fusion_dependency <- ccle_dependency_probability %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_dependency)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#Create dependency matrices for cell lines that have the fusion and cell lines that don't  
ccle_dependency_probability_matrix_with_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_with_fusion_dependency)

ccle_dependency_probability_matrix_without_fusion <- ccle_dependency_probability_matrix %>% subset(select = cell_lines_without_fusion_dependency)

#Compare dependency between the two groups with two sample t-test with equal variance and correct p-values using Benjamini-Hochberg procedure for multiple hypothesis testing. Also categorize by quartile of differential expression (joined from above). Then create an abstracted binary matrix
dependency_probability_with_vs_without_stats_TSTATS_null <- row_t_equalvar(ccle_dependency_probability_matrix_with_fusion, ccle_dependency_probability_matrix_without_fusion) %>% dplyr::mutate(gene = ccle_dependency_probability_matrix_df$gene) %>% dplyr::rename(dependency_probability_difference = `mean.diff`, dependency_probability_with_fusion = `mean.x`, dependency_probability_without_fusion = `mean.y`) %>% dplyr::select(gene, dependency_probability_difference, dependency_probability_with_fusion, dependency_probability_without_fusion, pvalue, statistic)

cell_permutation_matrices_TSTATS_1000[[number_of_cells]][iteration,] <- dependency_probability_with_vs_without_stats_TSTATS_null$statistic
                                                                 
count <- count + 1
print(count)

}

}


```

#Fusion-associated dependency t-statistic analysis: 3) Testing only partner genes and other TAD genes for each fusion, use the permutation-based t-statistics and actual t-statistics to determining FDRs for each possible fusion-associated dependency
```{r}

fusion_count_of_cells_with_dependency_data <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(DepMap_ID %in% colnames(ccle_dependency_probability_matrix)) %>% group_by(Fusion) %>% summarise(cell_count = n_distinct(DepMap_ID))

fusion_associated_dependency_TSTAT_FDR <- tibble(fusion = NA, dependency = NA, FDR = NA)
count <- 0

for(fusion in fusion_count_of_cells_with_dependency_data$Fusion){

number_of_cells <- fusion_count_of_cells_with_dependency_data %>% filter(Fusion == fusion) %>% select(cell_count) %>% unlist() %>% unname()
    
associated_gene_set <- intersect(cell_permutation_matrices_TSTATS_1000[[number_of_cells]] %>% colnames(), c(left_gene_TAD_neighbors[[fusion]], right_gene_TAD_neighbors[[fusion]],fusion_partners[[fusion]]))

if(associated_gene_set %>% length() > 0){

if(associated_gene_set %>% length() == 1){
permutation_tstats_matrix_for_associated_genes <- cbind(cell_permutation_matrices_TSTATS_1000[[number_of_cells]][,associated_gene_set],NA)  
colnames(permutation_tstats_matrix_for_associated_genes) <- c(associated_gene_set,NA)
}        
else{    
permutation_tstats_matrix_for_associated_genes <- cell_permutation_matrices_TSTATS_1000[[number_of_cells]][,associated_gene_set]
}

actuat_tstats_for_associated_genes <- dependency_probability_fusion_with_vs_without_TSTATS[[fusion]] %>% filter(gene %in% associated_gene_set)

for(dependency in associated_gene_set){
  
dependency_tstat <- dependency_probability_fusion_with_vs_without_TSTATS[[fusion]] %>% filter(gene == dependency) %>% select(statistic) %>% unlist() %>% unname()

number_of_hits_at_dependency_tstat_across_permutations <- c()

for(iteration in c(1:1000)){

permutation_tstats_matrix_for_associated_genes_iteration_with_NA_removed <- permutation_tstats_matrix_for_associated_genes[iteration,][!is.na(permutation_tstats_matrix_for_associated_genes[iteration,])]    
  
number_of_hits_at_dependency_tstat_across_permutations <- c(number_of_hits_at_dependency_tstat_across_permutations, permutation_tstats_matrix_for_associated_genes_iteration_with_NA_removed[permutation_tstats_matrix_for_associated_genes_iteration_with_NA_removed >= dependency_tstat] %>% length())

}

mean_number_of_hits_at_dependency_tstat_across_permutations <- number_of_hits_at_dependency_tstat_across_permutations %>% mean()

number_of_hits_at_dependency_tstat_for_fusion <- actuat_tstats_for_associated_genes$statistic[actuat_tstats_for_associated_genes$statistic >= dependency_tstat] %>% na.omit() %>% length()

FDR <- (mean_number_of_hits_at_dependency_tstat_across_permutations/number_of_hits_at_dependency_tstat_for_fusion)

fusion_associated_dependency_TSTAT_FDR <- add_row(fusion_associated_dependency_TSTAT_FDR, fusion = fusion, dependency = dependency, FDR = FDR)
  
}

}

count = count + 1
print(count)

}

fusion_associated_dependency_TSTAT_FDR_NA_removed <- fusion_associated_dependency_TSTAT_FDR %>% remove_missing()

#Write output of fusion-dependency pairings with associated FDRs from t-statistic analysis
write_tsv(fusion_associated_dependency_TSTAT_FDR_NA_removed, here::here("results/datatables","fusion_associated_dependency_TSTAT_FDR_NA_removed.tsv"))

```

#Fusion-associated dependency t-statistic analysis: 4) Comparison of results to original genome-scale analysis
```{r}

#Determining the FDRs according to t-statistics for each of the fusion-associated dependencies previously determine by genome-scale screening
fusion_associated_dependencies_with_FDRs_TSTATS <- TAD_signficant_dependency_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed %>% distinct(Fusion,Gene,Cell_Count) %>% select(Fusion,Gene,Cell_Count) %>% mutate(Fusion_Dependency_Pairing = str_c(Fusion,"_",Gene))

fusion_associated_dependencies_with_FDRs_TSTATS <- left_join(fusion_associated_dependencies_with_FDRs_TSTATS,fusion_associated_dependency_TSTAT_FDR_NA_removed, by = c("Fusion" = "fusion","Gene" = "dependency", "Fusion_Dependency_Pairing" = "Fusion_Dependency_Pairing"))

fusion_associated_dependencies_with_FDRs_TSTATS %>% view()

#Filtering fusion-associated depenendencies by various FDR thresholds according to t-statistics in order to enable comparison of results to genome-scale screening
fusion_associated_dependency_TSTAT_FDR_NA_removed <- fusion_associated_dependency_TSTAT_FDR_NA_removed %>% mutate(Fusion_Dependency_Pairing = str_c(fusion,"_",dependency))

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1 <- fusion_associated_dependency_TSTAT_FDR_NA_removed %>% filter(FDR < 0.1)

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1_duplicates_marked <- fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1 %>% mutate(fusion_inverse = str_c(str_split_fixed(fusion,"-",2)[,2],"-",str_split_fixed(fusion,"-",2)[,1])) %>% mutate(duplicate = fusion_inverse %in% fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1$fusion)

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1_duplicates_removed <- fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1_duplicates_marked %>% filter(!fusion %in% c("ABL1-BCR", "AFF1-KMT2A", "C2orf69-ROCK2", "CRYBG1-PDSS2", "MLLT10-PICALM", "NSD2-IGH", "RARA-PML"))

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05 <- fusion_associated_dependency_TSTAT_FDR_NA_removed %>% filter(FDR < 0.05)

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05_duplicates_marked <- fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05 %>% mutate(fusion_inverse = str_c(str_split_fixed(fusion,"-",2)[,2],"-",str_split_fixed(fusion,"-",2)[,1])) %>% mutate(duplicate = fusion_inverse %in% fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05$fusion)

fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05_duplicates_removed <- fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05_duplicates_marked %>% filter(!fusion %in% c("ABL1-BCR", "RARA-PML"))

```

#Orthogonal WGS data providing additional evidence for the presence of fusions
```{r}

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data <- ALL_fusions_protein_coding_filtered_list_final %>% filter(CCLE_Name %in% CCLE_translocations_SvABA_20181221$CCLE_name & Fusion %in% distinct_fusion_dependency_pairings$Fusion) %>% mutate(left_gene = str_split_fixed(Fusion,"-",2)[,1], right_gene = str_split_fixed(Fusion,"-",2)[,2]) %>% distinct(Fusion, CCLE_Name, left_gene, right_gene) %>% arrange(Fusion)

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_distinct_CCLE_Names <- list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data %>% distinct(CCLE_Name) %>% unlist() %>% unname()

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_by_cell_line <- list()
CCLE_translocations_SvABA_20181221_by_cell_line <- list()
counter = 0

for(cell_line in list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_distinct_CCLE_Names){
  
CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]] <- CCLE_translocations_SvABA_20181221 %>% filter(CCLE_name == cell_line)

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_by_cell_line[[cell_line]] <- list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data %>% filter(CCLE_Name == cell_line) %>% mutate(evidence_of_SV_in_WGS_data = ifelse(Fusion %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$fusion | Fusion %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$fusion_inverse,"Fusion seen",ifelse((left_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene1 | left_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene2) & (right_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene1 | right_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene2),"Both genes involved in SV",ifelse(left_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene1 | left_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene2,"Left gene involved in SV",ifelse(right_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene1 | right_gene %in% CCLE_translocations_SvABA_20181221_by_cell_line[[cell_line]]$gene2,"Right gene involved in SV","No WGS evidence")))))

counter = counter + 1
print(counter)
}

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined <- bind_rows(list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_by_cell_line)

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary <- list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined %>% group_by(evidence_of_SV_in_WGS_data) %>% summarise(count = n()) 

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary <- list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary %>% mutate(percentage = count/sum(list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary$count)*100)

```

#TCGA data from clinical samples to analyze fusions/ partners that have clinical correlates from this dataset
```{r}

#Identifying fusions from cell lines derived from tumors that have representation in the TCGA fusion call list
diseases_for_fusions_with_associated_dependencies <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion %in% unique(distinct_fusion_dependency_pairings$Fusion)) %>% distinct(Fusion, lineage, lineage_subtype) %>% arrange(Fusion)

tcga_disease_represented <- left_join(tcga_fusion_list_symbols_updated %>% distinct(Cancer), tcga_study_abbreviations, by = c("Cancer" = "Abbreviation"))

write_tsv(diseases_for_fusions_with_associated_dependencies, here::here("results/datatables","diseases_for_fusions_with_associated_dependencies.txt"))

write_tsv(tcga_disease_represented, here::here("results/datatables","tcga_disease_represented.txt"))

diseases_for_fusions_with_associated_dependencies_annotated <- read_tsv(here::here("results/datatables","diseases_for_fusions_with_associated_dependencies_annotated.txt"))

fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list <- diseases_for_fusions_with_associated_dependencies_annotated %>% filter(disease_in_tcga == "yes") %>% distinct(Fusion) %>% unlist() %>% unname()

#Summary tables to reference
tcga_fusion_summary <- tcga_fusion_list_symbols_updated %>% group_by(Fusion) %>% summarise(count = n_distinct(Sample)) %>% arrange(desc(count))

tcga_leftgene_summary <- tcga_fusion_list_symbols_updated %>% group_by(LeftGene) %>% summarise(count = n_distinct(Sample))

tcga_rightgene_summary <- tcga_fusion_list_symbols_updated %>% group_by(RightGene) %>% summarise(count = n_distinct(Sample))

tcga_partner_summary <- full_join(tcga_leftgene_summary, tcga_rightgene_summary, by = c("LeftGene" = "RightGene"))

tcga_partner_summary$count.x <- tcga_partner_summary$count.x %>% replace_na(0)

tcga_partner_summary$count.y <- tcga_partner_summary$count.y %>% replace_na(0)

tcga_partner_summary <- tcga_partner_summary %>% mutate(count = count.x + count.y) %>% select(gene = LeftGene, count)

#Dataframe of fusions with associated dependencies that show up themselves, or have partners that show up, in the TCGA dataset

frequency_of_fusion_and_partner_recurrence_in_human_samples <- tibble(Fusion = distinct_fusion_dependency_pairings$Fusion %>% unique()) %>% mutate(LeftGene = str_split_fixed(Fusion,"-",2)[,1], RightGene = str_split_fixed(Fusion,"-",2)[,2])

frequency_of_fusion_and_partner_recurrence_in_human_samples <- left_join(frequency_of_fusion_and_partner_recurrence_in_human_samples,tcga_fusion_summary, by = "Fusion") %>% rename(tcga_fusion_count = count)

frequency_of_fusion_and_partner_recurrence_in_human_samples <- left_join(frequency_of_fusion_and_partner_recurrence_in_human_samples,tcga_partner_summary, by = c("LeftGene" = "gene")) %>% rename(tcga_leftgene_count = count)

frequency_of_fusion_and_partner_recurrence_in_human_samples <- left_join(frequency_of_fusion_and_partner_recurrence_in_human_samples,tcga_partner_summary, by = c("RightGene" = "gene")) %>% rename(tcga_rightgene_count = count)

frequency_of_fusion_and_partner_recurrence_in_human_samples[is.na(frequency_of_fusion_and_partner_recurrence_in_human_samples)] <- 0

frequency_of_fusion_and_partner_recurrence_in_human_samples <- frequency_of_fusion_and_partner_recurrence_in_human_samples %>% mutate(tcga_max_partner_count = ifelse(tcga_leftgene_count > tcga_rightgene_count, tcga_leftgene_count, tcga_rightgene_count))

frequency_of_fusion_and_partner_recurrence_in_human_samples <- frequency_of_fusion_and_partner_recurrence_in_human_samples %>% mutate(tcga_fusion_count_group = ifelse(tcga_fusion_count == 0, "0",ifelse(tcga_fusion_count == 1, "1", ifelse(tcga_fusion_count > 1 & tcga_fusion_count <= 5,"2 - 5",ifelse(tcga_fusion_count > 5,"5+","error"))))) %>% mutate(tcga_max_partner_count_group = ifelse(tcga_max_partner_count == 0, "0",ifelse(tcga_max_partner_count == 1, "1", ifelse(tcga_max_partner_count > 1 & tcga_max_partner_count <= 5,"2 - 5",ifelse(tcga_max_partner_count > 5,"5+","error")))))

frequency_of_fusion_and_partner_recurrence_in_human_samples_filtered_for_fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list <- frequency_of_fusion_and_partner_recurrence_in_human_samples %>% filter(Fusion %in% fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list)

#Summary tables

summary_of_tcga_fusion_count_group <- frequency_of_fusion_and_partner_recurrence_in_human_samples %>% group_by(tcga_fusion_count_group) %>% summarise(count = n_distinct(Fusion)) %>% mutate(percent = count/sum(count)*100)

summary_of_tcga_max_partner_count_group <- frequency_of_fusion_and_partner_recurrence_in_human_samples %>% group_by(tcga_max_partner_count_group) %>% summarise(count = n_distinct(Fusion)) %>% mutate(percent = count/sum(count)*100)

summary_of_tcga_samples <- full_join(summary_of_tcga_fusion_count_group, summary_of_tcga_max_partner_count_group, by = c("tcga_fusion_count_group" = "tcga_max_partner_count_group")) %>% rename(frequency = tcga_fusion_count_group, fusion_percent = percent.x, partner_percent = percent.y) %>% select(frequency, fusion_percent, partner_percent)

summary_of_tcga_fusion_count_group_fusions_with_tcga_disease_rep <- frequency_of_fusion_and_partner_recurrence_in_human_samples_filtered_for_fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list %>% group_by(tcga_fusion_count_group) %>% summarise(count = n_distinct(Fusion)) %>% mutate(percent = count/sum(count)*100)

summary_of_tcga_max_partner_count_group_fusions_with_tcga_disease_rep <- frequency_of_fusion_and_partner_recurrence_in_human_samples_filtered_for_fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list %>% group_by(tcga_max_partner_count_group) %>% summarise(count = n_distinct(Fusion)) %>% mutate(percent = count/sum(count)*100)

summary_of_tcga_samples_fusions_with_tcga_disease_rep <- full_join(summary_of_tcga_fusion_count_group_fusions_with_tcga_disease_rep, summary_of_tcga_max_partner_count_group_fusions_with_tcga_disease_rep, by = c("tcga_fusion_count_group" = "tcga_max_partner_count_group")) %>% rename(frequency = tcga_fusion_count_group, fusion_percent = percent.x, partner_percent = percent.y) %>% select(frequency, fusion_percent, partner_percent)

frequency_of_fusion_and_partner_recurrence_in_human_samples_filtered_for_fusions_from_cell_lines_derived_from_tumors_with_representation_in_tcga_fusion_call_list %>% view()

```

#Orthogonal copy number and mutation data for fusion-associated dependencies
```{r}
copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- tibble(gene = NA, fusion = NA, cell_lines = NA, CCLE_names = NA, variant_in_dependency = NA, variant_classification = NA, is_deleterious = NA, variant_annotation = NA, copy_number_log2 = NA, loss_gain_neutral = NA)

counter = 0

for(gene in unique(distinct_fusion_dependency_pairings$Gene)){
  
fusions_with_gene_as_dependency <- distinct_fusion_dependency_pairings %>% filter(Gene == gene) %>% distinct(Fusion) %>% unlist() %>% unname()

for(fusion in fusions_with_gene_as_dependency){
  
cell_lines_with_fusion = ALL_fusions_protein_coding_filtered_list_final %>% filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname()  

CCLE_names_with_fusion = ALL_fusions_protein_coding_filtered_list_final %>% filter(Fusion == fusion) %>% distinct(CCLE_Name) %>% unlist() %>% unname()

cell_lines_with_each_gene_and_fusion <- tibble(cell_lines = cell_lines_with_fusion, CCLE_names = CCLE_names_with_fusion)
  
for(cell in cell_lines_with_each_gene_and_fusion$cell_lines){
  
  if(cell %in% CCLE_mutations$DepMap_ID){
    
    variants_in_cell = CCLE_mutations %>% filter(DepMap_ID == cell) %>% distinct(Hugo_Symbol) %>% unlist() %>% unname()

  } else{
  
    variants_in_cell = NA
}

variant_in_dependency <- gene %in% variants_in_cell

variant_classification <- ifelse(variant_in_dependency, CCLE_mutations %>% filter(DepMap_ID == cell & Hugo_Symbol == gene) %>% distinct(Variant_Classification) %>% unlist() %>% unname(), NA)

is_deleterious <- ifelse(variant_in_dependency, CCLE_mutations %>% filter(DepMap_ID == cell & Hugo_Symbol == gene) %>% distinct(isDeleterious) %>% unlist() %>% unname(), NA)

variant_annotation <- ifelse(variant_in_dependency, CCLE_mutations %>% filter(DepMap_ID == cell & Hugo_Symbol == gene) %>% distinct(Variant_annotation) %>% unlist() %>% unname(), NA)

copy_number_log2 <- ifelse(cell %in% CCLE_gene_cn$DepMap_ID, CCLE_gene_cn %>% filter(DepMap_ID == cell) %>% select(gene) %>% unlist() %>% unname(), NA)
  
copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- add_row(copy_number_and_mutation_data_by_dependency_fusion_and_cell_line,gene = gene, fusion = fusion, cell_lines = cell, CCLE_names = cell_lines_with_each_gene_and_fusion %>% filter(cell_lines == cell) %>% select(CCLE_names) %>% unlist() %>% unname(), variant_in_dependency = variant_in_dependency, variant_classification = variant_classification, is_deleterious = is_deleterious, variant_annotation = variant_annotation, copy_number_log2 = copy_number_log2, loss_gain_neutral = ifelse(copy_number_log2 > 1.5,"gain",ifelse(copy_number_log2 < 0.5,"loss",ifelse(is.na(copy_number_log2),NA,"neutral"))))
  
}

}

counter = counter + 1
print(counter)

}

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(dependency_data_available = cell_lines %in% colnames(ccle_dependency_probability_matrix))

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(is_partner = gene == str_split_fixed(fusion,"-",2)[,1] | gene == str_split_fixed(fusion,"-",2)[,2])

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(left_gene = str_split_fixed(fusion,"-",2)[,1], right_gene = str_split_fixed(fusion,"-",2)[,2])

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(cosmic_fusion = fusion %in% cosmic_fusions$value, left_gene_cosmic_translocation_partner = left_gene %in% cosmic_translocation_genes$`Gene Symbol`, right_gene_cosmic_translocation_partner = right_gene %in% cosmic_translocation_genes$`Gene Symbol`)

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(left_gene_kinase = left_gene %in% oncogenes_tsps_kinases_drivers$Kinase, right_gene_kinase = right_gene %in% oncogenes_tsps_kinases_drivers$Kinase)

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(left_gene_cosmic_census_gene = left_gene %in% cosmic_census_genes$`Gene Symbol`, right_gene_cosmic_census_gene = right_gene %in% cosmic_census_genes$`Gene Symbol`)

copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% rename(dependency = gene) %>% mutate(dependency_kinase = dependency %in% oncogenes_tsps_kinases_drivers$Kinase, dependency_cosmic_census_gene = dependency %in% cosmic_census_genes$`Gene Symbol`)


copy_number_alterations_involving_fusion_associated_dependencies_summary <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% group_by(is_partner,loss_gain_neutral) %>% summarise(count = n_distinct(dependency,fusion,cell_lines,CCLE_names))

copy_number_alterations_involving_fusion_associated_dependencies_summary <- copy_number_alterations_involving_fusion_associated_dependencies_summary %>% mutate(dependency_type = ifelse(is_partner == FALSE, "Collateral\ndependency", "Partner\ndependency"))

copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral <- copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral %>% replace_na("no data")

mutations_involving_fusion_associated_dependencies_summary <- copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% group_by(is_partner,variant_in_dependency) %>% summarise(count = n_distinct(dependency,fusion,cell_lines,CCLE_names))

mutations_involving_fusion_associated_dependencies_summary <- mutations_involving_fusion_associated_dependencies_summary %>% mutate(dependency_type = ifelse(is_partner == FALSE, "Collateral\ndependency", "Partner\ndependency"))

mutations_involving_fusion_associated_dependencies_summary <- mutations_involving_fusion_associated_dependencies_summary %>% select(dependency_type, variant_in_dependency, count)

```

#Differential RNA expression for fusion-associated dependencies
```{r}
RNA_expression_data_by_dependency_and_fusion <- tibble(gene = character(), fusion = character(), log2_fold_change = double(), log2_expression_with_fusion = double(), log2_expression_without_fusion = double(), pvalue = double())

counter = 0

for(fusion in unique(distinct_fusion_dependency_pairings$Fusion)){
  
dependencies_associated_with_fusion <- distinct_fusion_dependency_pairings %>% filter(Fusion == fusion) %>% distinct(Gene) %>% unlist() %>% unname()

#Cell lines with expression data that have fusion of interest and those that don't
cell_lines_with_fusion_expression <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_expression_protein_coding_log2tpm_matrix))
cell_lines_without_fusion_expression <- ccle_expression_protein_coding_log2tpm %>% dplyr::filter(!(DepMap_ID %in% cell_lines_with_fusion_expression)) %>% distinct(DepMap_ID) %>% unlist() %>% unname()

#Create expression matrices for cell lines that have the fusion and cell lines that don't
ccle_expression_matrix_with_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_with_fusion_expression)
ccle_expression_matrix_without_fusion <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = cell_lines_without_fusion_expression)

#Expression dataframe filtered for cell lines without fusion, for later use in density distribution
ccle_expression_protein_coding_log2tpm_without_fusion <- ccle_expression_protein_coding_log2tpm %>% filter(DepMap_ID %in% cell_lines_without_fusion_expression)

#Compare expression between the two groups with two sample t-test with equal variance
fusion_expression_table <- row_t_equalvar(ccle_expression_matrix_with_fusion, ccle_expression_matrix_without_fusion) %>% dplyr::mutate(gene = ccle_expression_protein_coding_log2tpm_matrix_df$gene) %>% dplyr::rename(log2_fold_change = `mean.diff`, log2_expression_with_fusion = `mean.x`, log2_expression_without_fusion = `mean.y`) %>% dplyr::select(gene, log2_fold_change, log2_expression_with_fusion, log2_expression_without_fusion, pvalue)

for(dependency in dependencies_associated_with_fusion){
  
if(dependency %in% colnames(ccle_expression_protein_coding_log2tpm_without_fusion)){

fusion_expression_table_for_dependency <- fusion_expression_table %>% filter(gene == dependency) %>% mutate(fusion = fusion) %>% select(gene,fusion,log2_fold_change,log2_expression_with_fusion,log2_expression_without_fusion,pvalue)

RNA_expression_data_by_dependency_and_fusion <- full_join(RNA_expression_data_by_dependency_and_fusion, fusion_expression_table_for_dependency)

}
  
}

counter = counter + 1
print(counter)

}

RNA_expression_data_by_dependency_and_fusion <- RNA_expression_data_by_dependency_and_fusion %>% rename(dependency = gene)
```

#Combined copy number, mutation, and RNA expression data for fusion-associated dependencies by cell line (while copy number and mutation data is unique to each cell line, RNA expression data is in aggregate for all the cell lines that have the fusion-dependency pairing)
```{r}

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- full_join(copy_number_and_mutation_data_by_dependency_fusion_and_cell_line, RNA_expression_data_by_dependency_and_fusion, by = c("dependency","fusion"))

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line[-1,]

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% filter(dependency_data_available == TRUE)

#Write output of RNA expression, copy number, mutation, and annotation data for fusion-dependency pairing
write_tsv(RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line, here::here("results/datatables","RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line.tsv"))

#Data for fusion-dependency co-description visualization
RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% mutate(cell_line_name = str_split_fixed(CCLE_names,"_",2)[,1])

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut$variant_classification <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut$variant_classification %>% replace_na("None")

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut$variant_classification <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut$variant_classification %>% recode("Missense_Mutation" = "Missense", "Nonsense_Mutation" = "Nonsense", "In_Frame_Del" = "In Frame Deletion", "Splice_Site" = "Splice Site")

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut <- left_join(RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut, disease_annotations_by_cell_line, by = c("cell_lines" = "DepMap_ID"))

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut %>% mutate(rna_overexpression = log2_fold_change > 1)

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut<- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut %>% rename(mutation = variant_classification, copy_number_alteration = loss_gain_neutral)

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut<- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut %>% rename(dependency_is_partner = is_partner)

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut_simplified <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut %>% select(dependency, fusion, cell_line_name, dependency_is_partner, cosmic_fusion, left_gene_cosmic_census_gene, right_gene_cosmic_census_gene, left_gene_kinase, right_gene_kinase, dependency_cosmic_census_gene, dependency_kinase, mutation, copy_number_alteration, rna_overexpression)

write_tsv(RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut_simplified, here::here("results/datatables","RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut_simplified.tsv"))

fusion_dependency_co_descriptor_plot <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line_formatted_for_CoMut_simplified %>% select(-c(mutation,copy_number_alteration,rna_overexpression))

fusion_dependency_co_descriptor_plot <- fusion_dependency_co_descriptor_plot %>% mutate(dependency_cosmic_census_gene = ifelse(dependency_cosmic_census_gene == TRUE, dependency, NA), dependency_kinase = ifelse(dependency_kinase == TRUE, dependency, NA))

fusion_dependency_co_descriptor_plot <- fusion_dependency_co_descriptor_plot %>% group_by(fusion) %>% summarise(cosmic_fusion = mean(cosmic_fusion), left_gene_cosmic_census_gene = mean(left_gene_cosmic_census_gene), right_gene_cosmic_census_gene = mean(right_gene_cosmic_census_gene), left_gene_kinase = mean(left_gene_kinase), right_gene_kinase = mean(right_gene_kinase), dependency_cosmic_census_gene = n_distinct(dependency_cosmic_census_gene,na.rm = TRUE), dependency_kinase = n_distinct(dependency_kinase,na.rm = TRUE))

fusion_dependency_co_descriptor_plot <- fusion_dependency_co_descriptor_plot %>% mutate(fusion_partner_cosmic_census_gene = left_gene_cosmic_census_gene + right_gene_cosmic_census_gene, fusion_partner_kinase = left_gene_kinase + right_gene_kinase)

fusion_dependency_co_descriptor_plot <- fusion_dependency_co_descriptor_plot %>% select(-c(left_gene_cosmic_census_gene,right_gene_cosmic_census_gene,left_gene_kinase,right_gene_kinase))

write_tsv(fusion_dependency_co_descriptor_plot, here::here("results/datatables","fusion_dependency_co_descriptor_plot.tsv"))

fusion_dependency_co_descriptor_plot_cosmic_annotation <- fusion_dependency_co_descriptor_plot %>% mutate(cosmic_fusion = ifelse(cosmic_fusion == 1, "COSMIC fusion", "Other fusion"))

fusion_dependency_co_descriptor_plot_cosmic_annotation_summary <- fusion_dependency_co_descriptor_plot_cosmic_annotation %>% group_by(cosmic_fusion) %>% summarise(dependency_cosmic_census_gene_percent = (dependency_cosmic_census_gene[!dependency_cosmic_census_gene == 0] %>% length())/(n_distinct(fusion))*100, dependency_kinase_percent = (dependency_kinase[!dependency_kinase == 0] %>% length())/(n_distinct(fusion))*100, fusion_partner_cosmic_census_gene_percent = (fusion_partner_cosmic_census_gene[!fusion_partner_cosmic_census_gene == 0] %>% length())/(n_distinct(fusion))*100, fusion_partner_kinase_percent = (fusion_partner_kinase[!fusion_partner_kinase == 0] %>% length())/(n_distinct(fusion))*100)

```

#sgRNA location for fusions with partner dependencies
```{r}

fusion_guide_information <- tibble(Fusion=NA, cell_lines=NA, CCLE_names=NA, left_gene_guide_estimated_to_cut=NA,left_gene_guide_total=NA,left_gene_guide_percentage_cutting=NA,right_gene_guide_estimated_to_cut=NA,right_gene_guide_total=NA,right_gene_guide_percentage_cutting=NA)

fusions_with_partner_dependency <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% filter(is_partner == TRUE) %>% select(fusion) %>% unlist() %>% unname()

counter = 0

for(fusion in fusions_with_partner_dependency){

left_gene_fusion <- str_split_fixed(fusion,"-",2)[,1]
right_gene_fusion <- str_split_fixed(fusion,"-",2)[,2]

cell_lines_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

CCLE_names_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% filter(DepMap_ID %in% cell_lines_with_fusion) %>% distinct(CCLE_Name) %>% unlist() %>% unname()

cell_lines_with_each_fusion <- tibble(cell_lines = cell_lines_with_fusion, CCLE_names = CCLE_names_with_fusion)
  
for(cell in cell_lines_with_each_fusion$cell_lines){
  
left_gene_vector_guides_cutting_estimate <- list()
left_gene_guide_estimated_to_cut <- list()
right_gene_vector_guides_cutting_estimate <- list()
right_gene_guide_estimated_to_cut <- list()

#If dependency data available on left gene
if(left_gene_fusion %in% ccle_dependency_probability_matrix_df$gene){
  
left_gene_orientation <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(LeftChromOrientation) %>% distinct() %>% unlist() %>% unname()

left_gene_vector_guide_locations <- ccle_dependency_guides %>% filter(gene == left_gene_fusion) %>% select(base) %>% distinct() %>% unlist() %>% unname() %>% as.double()

left_gene_breakpoints <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(LeftChromPosition) %>% unlist() %>% unname() %>% unique() %>% as.double()

left_gene_breakpoints_length <- left_gene_breakpoints %>% length()
  
#If left gene is in forward orientation, check to see if guides are to the left of the breakpoint and therefore estimated to cut
if(left_gene_orientation == "+"){
  
  for(breakpoint_index in c(1:left_gene_breakpoints_length)){
    
  left_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- left_gene_vector_guide_locations < left_gene_breakpoints[[breakpoint_index]]
  left_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(left_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
  
  }
#If left gene is in reverse orientation, check to see if guides are to the right of the breakpoint and therefore estimated to cut
  else{

  for(breakpoint_index in c(1:left_gene_breakpoints_length)){
  
  left_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- left_gene_vector_guide_locations > left_gene_breakpoints[[breakpoint_index]]
  left_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(left_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  } 
    
  }
  
}

#If dependency data available on right gene
if(right_gene_fusion %in% ccle_dependency_probability_matrix_df$gene){
  
right_gene_orientation <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(RightChromOrientation) %>% distinct() %>% unlist() %>% unname()

right_gene_vector_guide_locations <- ccle_dependency_guides %>% filter(gene == right_gene_fusion) %>% select(base) %>% distinct() %>% unlist() %>% unname() %>% as.double()

right_gene_breakpoints <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(RightChromPosition) %>% unlist() %>% unname() %>% unique() %>% as.double()

right_gene_breakpoints_length <- right_gene_breakpoints %>% length()
  
#If right gene is in forward orientation, check to see if guides are to the right of the breakpoint and therefore estimated to cut
if(right_gene_orientation == "+"){
  
  for(breakpoint_index in c(1:right_gene_breakpoints_length)){
  
  right_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- right_gene_vector_guide_locations > right_gene_breakpoints[[breakpoint_index]]
  right_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(right_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
   
  }
#If right gene is in reverse orientation, check to see if guides are to the left of the breakpoint and therefore estimated to cut
  else{

  for(breakpoint_index in c(1:right_gene_breakpoints_length)){
  
  right_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- right_gene_vector_guide_locations < right_gene_breakpoints[[breakpoint_index]]
  right_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(right_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
    
  }
  
}


left_gene_guide_total <- length(left_gene_vector_guide_locations)
left_gene_guide_percentage_cutting <- percent(max(left_gene_guide_estimated_to_cut %>% unlist())/left_gene_guide_total)

right_gene_guide_total <- length(right_gene_vector_guide_locations)
right_gene_guide_percentage_cutting <- percent(max(right_gene_guide_estimated_to_cut %>% unlist())/right_gene_guide_total)

fusion_guide_information <- add_row(fusion_guide_information, Fusion=fusion, cell_lines = cell, CCLE_names = cell_lines_with_each_fusion %>% filter(cell_lines == cell) %>% select(CCLE_names) %>% unlist() %>% unname(), left_gene_guide_estimated_to_cut=max(left_gene_guide_estimated_to_cut %>% unlist()), left_gene_guide_total=left_gene_guide_total,  left_gene_guide_percentage_cutting=left_gene_guide_percentage_cutting, right_gene_guide_estimated_to_cut=max(right_gene_guide_estimated_to_cut %>% unlist()), right_gene_guide_total=right_gene_guide_total, right_gene_guide_percentage_cutting=right_gene_guide_percentage_cutting)

}

counter = counter + 1
print(counter)

}

fusion_guide_information <- fusion_guide_information %>% mutate(dependency_data_available = cell_lines %in% colnames(ccle_dependency_probability_matrix)) %>% filter(dependency_data_available == TRUE)

is.na(fusion_guide_information) <- sapply(fusion_guide_information, is.infinite)

fusion_guide_information <- fusion_guide_information %>% select(Fusion, cell_lines, CCLE_names, left_gene_guide_estimated_to_cut, left_gene_guide_total, right_gene_guide_estimated_to_cut, right_gene_guide_total)

fusion_guide_information <- fusion_guide_information %>% distinct(Fusion, cell_lines, CCLE_names, left_gene_guide_estimated_to_cut, left_gene_guide_total, right_gene_guide_estimated_to_cut, right_gene_guide_total)

fusions_with_partner_dependencies_and_cosmic_status <- RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% filter(is_partner == TRUE & dependency_data_available == TRUE) %>% distinct(Fusion = fusion, cell_lines, CCLE_names, dependency, is_partner) %>% mutate(partner_dependency = ifelse(dependency == str_split_fixed(Fusion,"-",2)[,1],"left",ifelse(dependency == str_split_fixed(Fusion,"-",2)[,2],"right","none")))

fusion_guide_information_annotated_for_partner_dependencies <- left_join(fusions_with_partner_dependencies_and_cosmic_status, fusion_guide_information)

fusion_guide_information_annotated_for_partner_dependencies$partner_dependency <- fusion_guide_information_annotated_for_partner_dependencies$partner_dependency %>% replace_na("none")

fusion_guide_information_annotated_for_partner_dependencies <- fusion_guide_information_annotated_for_partner_dependencies %>% mutate(cosmic_fusion = Fusion %in% cosmic_fusions$value)

#Proportion of fusion-dependency-cell line combinations with partner dependencies that don't have sgRNAs falling on predicted fusion transcripts

partner_fusion_dependency_cell_line_combinations <- fusion_guide_information_annotated_for_partner_dependencies %>% summarise(n_distinct(Fusion,dependency,cell_lines)) %>% unlist() %>% unname()

partner_fusion_dependency_cell_line_combinations_with_sgRNAs_off_fusion_transcript <- fusion_guide_information_annotated_for_partner_dependencies %>% filter(partner_dependency == "left" & left_gene_guide_estimated_to_cut == 0) %>% summarise(n_distinct(Fusion,dependency,cell_lines)) %>% unlist() %>% unname() + fusion_guide_information_annotated_for_partner_dependencies %>% filter(partner_dependency == "right" & right_gene_guide_estimated_to_cut == 0) %>% summarise(n_distinct(Fusion,dependency,cell_lines)) %>% unlist() %>% unname()

proportion_of_partner_fusion_dependency_cell_line_combinations_with_sgRNAs_off_fusion_transcript <- partner_fusion_dependency_cell_line_combinations_with_sgRNAs_off_fusion_transcript/partner_fusion_dependency_cell_line_combinations

#Write output of fusion sgRNA locations
write_tsv(fusion_guide_information_annotated_for_partner_dependencies,here::here("results/datatables","fusion_guide_information_annotated_for_partner_dependencies.tsv"))

```

#CTD2 drug screening data for multiple myeloma cells
```{r}

#Add column identifying kinase targets to compound metadata

ctd2_compound_metadata <- ctd2_compound_metadata %>% mutate(list_of_targets = (gene_symbol_of_protein_target %>% str_split(pattern = ";")))

return_any_targets_that_are_kinases <- function(targets){

return(intersect(targets, oncogenes_tsps_kinases_drivers$Kinase) %>% str_c(sep = ",", collapse = ", "))
  
}

ctd2_compound_metadata$kinase_targets <- lapply(ctd2_compound_metadata$list_of_targets, return_any_targets_that_are_kinases)

ctd2_compound_metadata <- ctd2_compound_metadata %>% mutate(kinase_targets = ifelse(kinase_targets == "character(0)",FALSE,kinase_targets))

ctd2_compound_metadata <- ctd2_compound_metadata %>% mutate(kinase_targets = kinase_targets %>% as.character())

#Combining AUC data with metadata and formatting a tibble with cell names in the first column as rows and drug names as columns

ctd2_auc_annotated <- full_join(ctd2_auc, ctd2_experiment_metadata, by = "experiment_id")

ctd2_auc_annotated <- full_join(ctd2_auc_annotated, ctd2_cell_metadata, by = "master_ccl_id")

ctd2_auc_annotated <- full_join(ctd2_auc_annotated, ctd2_compound_metadata, by = "master_cpd_id")

ctd2_auc_gather <- ctd2_auc_annotated %>% select(ccl_name, cpd_name, area_under_curve)

ctd2_auc_gather_mean <- ctd2_auc_gather %>% group_by(ccl_name, cpd_name) %>% summarise(area_under_curve = mean(area_under_curve))

ctd2_auc_spread_mean <- ctd2_auc_gather_mean %>% spread(key = cpd_name, value = area_under_curve) %>% select(-547)

ctd2_auc_spread_mean_matrix <- ctd2_auc_spread_mean %>% column_to_rownames("ccl_name") %>% t()

ctd2_auc_spread_mean_matrix_df <- ctd2_auc_spread_mean_matrix %>% as.data.frame() %>% rownames_to_column(var = "cpd_name") 

#Multiple myeloma cell lines

multiple_myeloma_cells_all <- ALL_fusions_protein_coding_filtered_list_final %>% filter(lineage_subtype == "multiple_myeloma") %>% distinct(stripped_cell_line_name) %>% unlist() %>% unname()

multiple_myeloma_cells_IGH_NSD2 <- ALL_fusions_protein_coding_filtered_list_final %>% filter(lineage_subtype == "multiple_myeloma" & Fusion == "IGH-NSD2") %>% distinct(stripped_cell_line_name) %>% unlist() %>% unname()

multiple_myeloma_cells_IGH_NSD2_CTD2_data <- intersect(multiple_myeloma_cells_IGH_NSD2,colnames(ctd2_auc_spread_mean_matrix))

multiple_myeloma_cells_IGH_NSD2_null <- setdiff(multiple_myeloma_cells_all,multiple_myeloma_cells_IGH_NSD2)

multiple_myeloma_cells_IGH_NSD2_null_CTD2_data <- intersect(multiple_myeloma_cells_IGH_NSD2_null,colnames(ctd2_auc_spread_mean_matrix))

#Create AUC matrices for cell lines that have the fusion and cell lines that don't  
ctd2_auc_spread_mean_matrix_multiple_myeloma_cells_IGH_NSD2 <- ctd2_auc_spread_mean_matrix %>% subset(select = multiple_myeloma_cells_IGH_NSD2_CTD2_data)

ctd2_auc_spread_mean_matrix_multiple_myeloma_cells_IGH_NSD2_null <- ctd2_auc_spread_mean_matrix %>% subset(select = multiple_myeloma_cells_IGH_NSD2_null_CTD2_data)

#Compare AUC between the two groups with two sample t-test with equal variance and correct p-values using Benjamini-Hochberg procedure for multiple hypothesis testing
ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats <- row_t_equalvar(ctd2_auc_spread_mean_matrix_multiple_myeloma_cells_IGH_NSD2, ctd2_auc_spread_mean_matrix_multiple_myeloma_cells_IGH_NSD2_null) %>% dplyr::mutate(BH_QValue = p.adjust(pvalue, "BH", length(ctd2_auc_spread_mean_matrix_multiple_myeloma_cells_IGH_NSD2[,1]))) %>% dplyr::mutate(negative_log10_BH_QValue = -log10(BH_QValue), cpd_name = ctd2_auc_spread_mean_matrix_df$cpd_name) %>% dplyr::rename(auc_difference = `mean.diff`, auc_with_fusion = `mean.x`, auc_without_fusion = `mean.y`) %>% dplyr::select(cpd_name, auc_difference, auc_with_fusion, auc_without_fusion, negative_log10_BH_QValue)

ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats_annotated <- full_join(ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats,ctd2_compound_metadata)

```

#RNA expression for multiple myeloma cells
```{r}

#Multiple myeloma cells with expression data
multiple_myeloma_cells_all_DepMap_ID <- ALL_fusions_protein_coding_filtered_list_final %>% filter(lineage_subtype == "multiple_myeloma") %>% distinct(DepMap_ID) %>% unlist() %>% unname()

multiple_myeloma_cells_IGH_NSD2_DepMap_ID <- ALL_fusions_protein_coding_filtered_list_final %>% filter(lineage_subtype == "multiple_myeloma" & Fusion == "IGH-NSD2") %>% distinct(DepMap_ID) %>% unlist() %>% unname()

multiple_myeloma_cells_IGH_NSD2_RNA_expression_data <- intersect(multiple_myeloma_cells_IGH_NSD2_DepMap_ID,colnames(ccle_expression_protein_coding_log2tpm_matrix))

multiple_myeloma_cells_IGH_NSD2_null_DepMap_ID <- setdiff(multiple_myeloma_cells_all_DepMap_ID,multiple_myeloma_cells_IGH_NSD2_DepMap_ID)

multiple_myeloma_cells_IGH_NSD2_null_RNA_expression_data <- intersect(multiple_myeloma_cells_IGH_NSD2_null_DepMap_ID,colnames(ccle_expression_protein_coding_log2tpm_matrix))

#Create RNA expression matrices for cell lines that have the fusion and cell lines that don't  
RNA_expression_matrix_multiple_myeloma_cells_IGH_NSD2 <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = multiple_myeloma_cells_IGH_NSD2_RNA_expression_data)

RNA_expression_mean_matrix_multiple_myeloma_cells_IGH_NSD2_null <- ccle_expression_protein_coding_log2tpm_matrix %>% subset(select = multiple_myeloma_cells_IGH_NSD2_null_RNA_expression_data)

#Compare RNA expression between the two groups with two sample t-test with equal variance and correct p-values using Benjamini-Hochberg procedure for multiple hypothesis testing
RNA_expression_myeloma_cells_IGH_NSD2_with_vs_without_stats <- row_t_equalvar(RNA_expression_matrix_multiple_myeloma_cells_IGH_NSD2, RNA_expression_mean_matrix_multiple_myeloma_cells_IGH_NSD2_null) %>% dplyr::mutate(BH_QValue = p.adjust(pvalue, "BH", length(RNA_expression_matrix_multiple_myeloma_cells_IGH_NSD2[,1]))) %>% dplyr::mutate(negative_log10_BH_QValue = -log10(BH_QValue), gene = ccle_expression_protein_coding_log2tpm_matrix_df$gene) %>% dplyr::rename(log2_fold_change = `mean.diff`, expression_with_fusion = `mean.x`, expression_without_fusion = `mean.y`) %>% dplyr::select(gene, log2_fold_change, expression_with_fusion, expression_without_fusion, negative_log10_BH_QValue)

```

#Guide placement for NSD2 in IGH-NSD2 fusion
```{r}

fusion_guide_information_for_select_fusion <- tibble(Fusion=NA, cell_lines=NA, CCLE_names=NA, left_gene_guide_estimated_to_cut=NA,left_gene_guide_total=NA,left_gene_guide_percentage_cutting=NA,right_gene_guide_estimated_to_cut=NA,right_gene_guide_total=NA,right_gene_guide_percentage_cutting=NA)

fusion = "IGH-NSD2"

left_gene_fusion <- str_split_fixed(fusion,"-",2)[,1]
right_gene_fusion <- str_split_fixed(fusion,"-",2)[,2]

cell_lines_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% dplyr::filter(Fusion == fusion) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

CCLE_names_with_fusion <- ALL_fusions_protein_coding_filtered_list_final %>% filter(DepMap_ID %in% cell_lines_with_fusion) %>% distinct(CCLE_Name) %>% unlist() %>% unname()

cell_lines_with_each_fusion <- tibble(cell_lines = cell_lines_with_fusion, CCLE_names = CCLE_names_with_fusion)
  
for(cell in cell_lines_with_each_fusion$cell_lines){
  
left_gene_vector_guides_cutting_estimate <- list()
left_gene_guide_estimated_to_cut <- list()
right_gene_vector_guides_cutting_estimate <- list()
right_gene_guide_estimated_to_cut <- list()

#If dependency data available on left gene
if(left_gene_fusion %in% ccle_dependency_probability_matrix_df$gene){
  
left_gene_orientation <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(LeftChromOrientation) %>% distinct() %>% unlist() %>% unname()

left_gene_vector_guide_locations <- ccle_dependency_guides %>% filter(gene == left_gene_fusion) %>% select(base) %>% distinct() %>% unlist() %>% unname() %>% as.double()

left_gene_breakpoints <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(LeftChromPosition) %>% unlist() %>% unname() %>% unique() %>% as.double()

left_gene_breakpoints_length <- left_gene_breakpoints %>% length()
  
#If left gene is in forward orientation, check to see if guides are to the left of the breakpoint and therefore estimated to cut
if(left_gene_orientation == "+"){
  
  for(breakpoint_index in c(1:left_gene_breakpoints_length)){
    
  left_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- left_gene_vector_guide_locations < left_gene_breakpoints[[breakpoint_index]]
  left_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(left_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
  
  }
#If left gene is in reverse orientation, check to see if guides are to the right of the breakpoint and therefore estimated to cut
  else{

  for(breakpoint_index in c(1:left_gene_breakpoints_length)){
  
  left_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- left_gene_vector_guide_locations > left_gene_breakpoints[[breakpoint_index]]
  left_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(left_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  } 
    
  }
  
}

#If dependency data available on right gene
if(right_gene_fusion %in% ccle_dependency_probability_matrix_df$gene){
  
right_gene_orientation <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(RightChromOrientation) %>% distinct() %>% unlist() %>% unname()

right_gene_vector_guide_locations <- ccle_dependency_guides %>% filter(gene == right_gene_fusion) %>% select(base) %>% distinct() %>% unlist() %>% unname() %>% as.double()

right_gene_breakpoints <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion == fusion & DepMap_ID == cell) %>% select(RightChromPosition) %>% unlist() %>% unname() %>% unique() %>% as.double()

right_gene_breakpoints_length <- right_gene_breakpoints %>% length()
  
#If right gene is in forward orientation, check to see if guides are to the right of the breakpoint and therefore estimated to cut
if(right_gene_orientation == "+"){
  
  for(breakpoint_index in c(1:right_gene_breakpoints_length)){
  
  right_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- right_gene_vector_guide_locations > right_gene_breakpoints[[breakpoint_index]]
  right_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(right_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
   
  }
#If right gene is in reverse orientation, check to see if guides are to the left of the breakpoint and therefore estimated to cut
  else{

  for(breakpoint_index in c(1:right_gene_breakpoints_length)){
  
  right_gene_vector_guides_cutting_estimate[[breakpoint_index]] <- right_gene_vector_guide_locations < right_gene_breakpoints[[breakpoint_index]]
  right_gene_guide_estimated_to_cut[[breakpoint_index]] <- sum(right_gene_vector_guides_cutting_estimate[[breakpoint_index]] == TRUE)
    
  }
    
  }
  
}


left_gene_guide_total <- length(left_gene_vector_guide_locations)
left_gene_guide_percentage_cutting <- percent(max(left_gene_guide_estimated_to_cut %>% unlist())/left_gene_guide_total)

right_gene_guide_total <- length(right_gene_vector_guide_locations)
right_gene_guide_percentage_cutting <- percent(max(right_gene_guide_estimated_to_cut %>% unlist())/right_gene_guide_total)

fusion_guide_information_for_select_fusion <- add_row(fusion_guide_information_for_select_fusion, Fusion=fusion, cell_lines = cell, CCLE_names = cell_lines_with_each_fusion %>% filter(cell_lines == cell) %>% select(CCLE_names) %>% unlist() %>% unname(), left_gene_guide_estimated_to_cut=max(left_gene_guide_estimated_to_cut %>% unlist()), left_gene_guide_total=left_gene_guide_total,  left_gene_guide_percentage_cutting=left_gene_guide_percentage_cutting, right_gene_guide_estimated_to_cut=max(right_gene_guide_estimated_to_cut %>% unlist()), right_gene_guide_total=right_gene_guide_total, right_gene_guide_percentage_cutting=right_gene_guide_percentage_cutting)

}

fusion_guide_information_for_select_fusion <- fusion_guide_information_for_select_fusion[-1,]

write_tsv(fusion_guide_information_for_select_fusion,here::here("results/datatables","fusion_guide_information_for_multiple_myeloma_cells_with_IGH_NSD2_fusion.tsv"))

```

#Analysis for CRISPR data from NCIH23 lung cancer cell based organoid model CRISPR screen from "CRISPR screens in cancer spheroids identify 3D growth-specific vulnerabilities". Looking to see if EML4, a collateral dependency in the cell line, is also "disenriched" in the organoid model
```{r}

#Analysis for EML4

#NCIH23

NCIH23_sgrna_counts <- full_join(NCIH23_D19_3D_U1_counts, NCIH23_D19_3D_U2_counts, by = "sgrna")

NCIH23_sgrna_counts <- full_join(NCIH23_sgrna_counts, NCIH23_T0_counts, by = "sgrna")

NCIH23_sgrna_counts <- NCIH23_sgrna_counts %>% rename(replicate_1 = count.x, replicate_2 = count.y, t_0 = count)

NCIH23_sgrna_counts <- NCIH23_sgrna_counts %>% mutate(gene = str_split_fixed(sgrna,"_",3)[,2]) %>% select(sgrna,gene,replicate_1,replicate_2,t_0)

NCIH23_sgrna_counts <- NCIH23_sgrna_counts %>% mutate(replicate_1_log2fc = log2(replicate_1/t_0), replicate_2_log2fc = log2(replicate_2/t_0))

NCIH23_median_negative_control_sgrna_replicate_1 <- NCIH23_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH23_sd_negative_control_sgrna_replicate_1 <- NCIH23_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH23_median_negative_control_sgrna_replicate_2 <- NCIH23_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH23_sd_negative_control_sgrna_replicate_2 <- NCIH23_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH23_sgrna_counts <- NCIH23_sgrna_counts %>% mutate(replicate_1_log2fc_normalized = replicate_1_log2fc - NCIH23_median_negative_control_sgrna_replicate_1, replicate_2_log2fc_normalized = replicate_2_log2fc - NCIH23_median_negative_control_sgrna_replicate_2)

NCIH23_sgrna_counts <- NCIH23_sgrna_counts %>% mutate(replicate_1_pZ = replicate_1_log2fc_normalized/NCIH23_sd_negative_control_sgrna_replicate_1, replicate_2_pZ = replicate_2_log2fc_normalized/NCIH23_sd_negative_control_sgrna_replicate_2)

NCIH23_sgrna_counts_pZ <- NCIH23_sgrna_counts %>% select(sgrna,gene,replicate_1_pZ,replicate_2_pZ)

NCIH23_sgrna_counts_pZ_gather <- NCIH23_sgrna_counts_pZ %>% gather(key = "replicate", value = "pZ", c(replicate_1_pZ, replicate_2_pZ))

NCIH23_sgrna_counts_pZ_gather_EML4 <- NCIH23_sgrna_counts_pZ_gather %>% filter(gene == "EML4") %>% mutate(cell_line = "NCIH23")

NCIH23_sgrna_counts_pZ_gather_EML4$replicate <- NCIH23_sgrna_counts_pZ_gather_EML4$replicate %>% recode(replicate_1_pZ = "Replicate 1", replicate_2_pZ = "Replicate 2")

#NCIH1975

NCIH1975_sgrna_counts <- full_join(NCIH1975_D19_3D_U1_counts, NCIH1975_D19_3D_U2_counts, by = "sgrna")

NCIH1975_sgrna_counts <- full_join(NCIH1975_sgrna_counts, NCIH1975_T0_counts, by = "sgrna")

NCIH1975_sgrna_counts <- NCIH1975_sgrna_counts %>% rename(replicate_1 = count.x, replicate_2 = count.y, t_0 = count)

NCIH1975_sgrna_counts <- NCIH1975_sgrna_counts %>% mutate(gene = str_split_fixed(sgrna,"_",3)[,2]) %>% select(sgrna,gene,replicate_1,replicate_2,t_0)

NCIH1975_sgrna_counts <- NCIH1975_sgrna_counts %>% mutate(replicate_1_log2fc = log2(replicate_1/t_0), replicate_2_log2fc = log2(replicate_2/t_0))

NCIH1975_median_negative_control_sgrna_replicate_1 <- NCIH1975_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH1975_sd_negative_control_sgrna_replicate_1 <- NCIH1975_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH1975_median_negative_control_sgrna_replicate_2 <- NCIH1975_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH1975_sd_negative_control_sgrna_replicate_2 <- NCIH1975_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH1975_sgrna_counts <- NCIH1975_sgrna_counts %>% mutate(replicate_1_log2fc_normalized = replicate_1_log2fc - NCIH1975_median_negative_control_sgrna_replicate_1, replicate_2_log2fc_normalized = replicate_2_log2fc - NCIH1975_median_negative_control_sgrna_replicate_2)

NCIH1975_sgrna_counts <- NCIH1975_sgrna_counts %>% mutate(replicate_1_pZ = replicate_1_log2fc_normalized/NCIH1975_sd_negative_control_sgrna_replicate_1, replicate_2_pZ = replicate_2_log2fc_normalized/NCIH1975_sd_negative_control_sgrna_replicate_2)

NCIH1975_sgrna_counts_pZ <- NCIH1975_sgrna_counts %>% select(sgrna,gene,replicate_1_pZ,replicate_2_pZ)

NCIH1975_sgrna_counts_pZ_gather <- NCIH1975_sgrna_counts_pZ %>% gather(key = "replicate", value = "pZ", c(replicate_1_pZ, replicate_2_pZ))

NCIH1975_sgrna_counts_pZ_gather_EML4 <- NCIH1975_sgrna_counts_pZ_gather %>% filter(gene == "EML4") %>% mutate(cell_line = "NCIH1975")

NCIH1975_sgrna_counts_pZ_gather_EML4$replicate <- NCIH1975_sgrna_counts_pZ_gather_EML4$replicate %>% recode(replicate_1_pZ = "Replicate 1", replicate_2_pZ = "Replicate 2")

#NCIH2009

NCIH2009_sgrna_counts <- full_join(NCIH2009_D19_3D_U1_counts, NCIH2009_D19_3D_U2_counts, by = "sgrna")

NCIH2009_sgrna_counts <- full_join(NCIH2009_sgrna_counts, NCIH2009_T0_counts, by = "sgrna")

NCIH2009_sgrna_counts <- NCIH2009_sgrna_counts %>% rename(replicate_1 = count.x, replicate_2 = count.y, t_0 = count)

NCIH2009_sgrna_counts <- NCIH2009_sgrna_counts %>% mutate(gene = str_split_fixed(sgrna,"_",3)[,2]) %>% select(sgrna,gene,replicate_1,replicate_2,t_0)

NCIH2009_sgrna_counts <- NCIH2009_sgrna_counts %>% mutate(replicate_1_log2fc = log2(replicate_1/t_0), replicate_2_log2fc = log2(replicate_2/t_0))

NCIH2009_median_negative_control_sgrna_replicate_1 <- NCIH2009_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH2009_sd_negative_control_sgrna_replicate_1 <- NCIH2009_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_1_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH2009_median_negative_control_sgrna_replicate_2 <- NCIH2009_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(median(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH2009_sd_negative_control_sgrna_replicate_2 <- NCIH2009_sgrna_counts %>% filter(gene %in% c("none","safe")) %>% summarise(sd(replicate_2_log2fc,na.rm = TRUE)) %>% unlist() %>% unname() %>% as.double()

NCIH2009_sgrna_counts <- NCIH2009_sgrna_counts %>% mutate(replicate_1_log2fc_normalized = replicate_1_log2fc - NCIH2009_median_negative_control_sgrna_replicate_1, replicate_2_log2fc_normalized = replicate_2_log2fc - NCIH2009_median_negative_control_sgrna_replicate_2)

NCIH2009_sgrna_counts <- NCIH2009_sgrna_counts %>% mutate(replicate_1_pZ = replicate_1_log2fc_normalized/NCIH2009_sd_negative_control_sgrna_replicate_1, replicate_2_pZ = replicate_2_log2fc_normalized/NCIH2009_sd_negative_control_sgrna_replicate_2)

NCIH2009_sgrna_counts_pZ <- NCIH2009_sgrna_counts %>% select(sgrna,gene,replicate_1_pZ,replicate_2_pZ)

NCIH2009_sgrna_counts_pZ_gather <- NCIH2009_sgrna_counts_pZ %>% gather(key = "replicate", value = "pZ", c(replicate_1_pZ, replicate_2_pZ))

NCIH2009_sgrna_counts_pZ_gather_EML4 <- NCIH2009_sgrna_counts_pZ_gather %>% filter(gene == "EML4") %>% mutate(cell_line = "NCIH2009")

NCIH2009_sgrna_counts_pZ_gather_EML4$replicate <- NCIH2009_sgrna_counts_pZ_gather_EML4$replicate %>% recode(replicate_1_pZ = "Replicate 1", replicate_2_pZ = "Replicate 2")

#Combined data table of phenotypic Z scores for sgRNAs targeting EML4

combined_sgrna_counts_pZ_gather_EML4 <- bind_rows(NCIH23_sgrna_counts_pZ_gather_EML4, NCIH1975_sgrna_counts_pZ_gather_EML4, NCIH2009_sgrna_counts_pZ_gather_EML4)

combined_sgrna_counts_pZ_gather_EML4 <- combined_sgrna_counts_pZ_gather_EML4 %>% mutate(cell_line_with_or_without_fusion = ifelse(cell_line == "NCIH23","Fusion", "No fusion"))

combined_sgrna_counts_pZ_gather_EML4$cell_line <- factor(combined_sgrna_counts_pZ_gather_EML4$cell_line, levels = c("NCIH23","NCIH1975", "NCIH2009"))

combined_sgrna_counts_pZ_gather_EML4$cell_line_with_or_without_fusion <- factor(combined_sgrna_counts_pZ_gather_EML4$cell_line_with_or_without_fusion, levels = c("Fusion","No fusion"))

#Combined data table of phenotypic Z scores for sgRNAs targeting all nonessential genes

NCIH23_sgrna_counts_pZ_gather <- NCIH23_sgrna_counts_pZ_gather %>% mutate(cell_line = "NCIH23")

NCIH1975_sgrna_counts_pZ_gather <- NCIH1975_sgrna_counts_pZ_gather %>% mutate(cell_line = "NCIH1975")

NCIH2009_sgrna_counts_pZ_gather <- NCIH2009_sgrna_counts_pZ_gather %>% mutate(cell_line = "NCIH2009")

combined_sgrna_counts_pZ_gather <- bind_rows(NCIH23_sgrna_counts_pZ_gather, NCIH1975_sgrna_counts_pZ_gather, NCIH2009_sgrna_counts_pZ_gather)

combined_sgrna_counts_pZ_gather <- combined_sgrna_counts_pZ_gather %>% mutate(cell_line_with_or_without_fusion = ifelse(cell_line == "NCIH23","Fusion", "No fusion"))

combined_sgrna_counts_pZ_gather$cell_line <- factor(combined_sgrna_counts_pZ_gather$cell_line, levels = c("NCIH23","NCIH1975", "NCIH2009"))

combined_sgrna_counts_pZ_gather$cell_line_with_or_without_fusion <- factor(combined_sgrna_counts_pZ_gather$cell_line_with_or_without_fusion, levels = c("Fusion","No fusion"))

combined_sgrna_counts_pZ_gather_all_nonessential <- combined_sgrna_counts_pZ_gather %>% filter(gene %in% nonessentials_gene_list)

#Mean phenotypic Z scores of nonessential genes (grouped by gene)

mean_pZ_nonessentials <- combined_sgrna_counts_pZ_gather_all_nonessential %>% group_by(cell_line, cell_line_with_or_without_fusion, gene) %>% summarize(mean_pZ = mean(pZ, na.rm = TRUE))

```

#Analysis of which fusion-dependency pairings fall into fragile sites
```{r}

#Distinct fusion-dependency pairings annotated for chromosome starts and ends (for some reason there are two duplicated records - seems like some right partners may appear in the gene table multiple times)
distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends <- distinct_fusion_dependency_pairings %>% rename(fusion = Fusion, dependency = Gene) %>% select(fusion, dependency) %>% mutate(left_partner = str_split_fixed(fusion,"-",2)[,1], right_partner = str_split_fixed(fusion,"-",2)[,2])

distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends <- left_join(distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends,gene_coordinate_table_hg38, by = c("dependency" = "hgnc_symbol")) %>% rename(dependency_chromosome = chromosome_name, dependency_start = start_position, dependency_end = end_position)

distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends <- left_join(distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends,gene_coordinate_table_hg38, by = c("left_partner" = "hgnc_symbol")) %>% rename(left_partner_chromosome = chromosome_name, left_partner_start = start_position, left_partner_end = end_position)

distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends <- left_join(distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends,gene_coordinate_table_hg38, by = c("right_partner" = "hgnc_symbol")) %>% rename(right_partner_chromosome = chromosome_name, right_partner_start = start_position, right_partner_end = end_position)

#Looking at partners and dependencies in fragile sites

sqldf('SELECT distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.*,fragile_sites_PCAWG_hg38.* FROM distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends INNER JOIN fragile_sites_PCAWG_hg38 ON ((distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.dependency_start BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end) OR (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.dependency_end BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end)) AND (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.dependency_chromosome = fragile_sites_PCAWG_hg38.chromosome)')

sqldf('SELECT distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.*,fragile_sites_PCAWG_hg38.* FROM distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends INNER JOIN fragile_sites_PCAWG_hg38 ON ((distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.left_partner_start BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end) OR (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.left_partner_end BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end)) AND (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.left_partner_chromosome = fragile_sites_PCAWG_hg38.chromosome)')

sqldf('SELECT distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.*,fragile_sites_PCAWG_hg38.* FROM distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends INNER JOIN fragile_sites_PCAWG_hg38 ON ((distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.right_partner_start BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end) OR (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.right_partner_end BETWEEN fragile_sites_PCAWG_hg38.start AND fragile_sites_PCAWG_hg38.end)) AND (distinct_fusion_dependency_pairings_annotated_for_chromosome_starts_and_ends.right_partner_chromosome = fragile_sites_PCAWG_hg38.chromosome)')

```

#Plots
```{r}

#100% stacked bar plot of all fusions and cell lines stratified by those that have partner dependencies, unique other TAD dependencies, or no clearly associated dependencies

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line$cell_lines %>% unique()

fusions_with_partner_dependency <- distinct_fusion_dependency_pairings %>% filter(is_partner == TRUE) %>% distinct(Fusion) %>% unlist() %>% unname()

fusions_with_unique_other_tad_dependency <- setdiff(distinct_fusion_dependency_pairings %>% filter(is_partner == FALSE) %>% distinct(Fusion) %>% unlist() %>% unname(), fusions_with_partner_dependency)

fusion_with_no_associated_dependency <- setdiff(ALL_fusions_with_dependency_data,c(fusions_with_partner_dependency,fusions_with_unique_other_tad_dependency))

total_fusions_with_dependency_data <- unique(ALL_fusions_with_dependency_data)


cell_lines_with_partner_dependency <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion %in% fusions_with_partner_dependency) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix))

cell_lines_with_unique_other_tad_dependency <- setdiff(ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(Fusion %in% fusions_with_unique_other_tad_dependency) %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix)), cell_lines_with_partner_dependency)

cell_lines_with_no_associated_dependency <- setdiff(ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% distinct(DepMap_ID) %>% unlist() %>% unname() %>% intersect(colnames(ccle_dependency_probability_matrix)),c(cell_lines_with_partner_dependency,cell_lines_with_unique_other_tad_dependency))

total_cell_lines_with_dependency_data <- c(cell_lines_with_partner_dependency,cell_lines_with_unique_other_tad_dependency,cell_lines_with_no_associated_dependency)

ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20 <- tibble(Fusion_or_Cell_Lines = c("Fusions","Fusions","Fusions","Cell lines","Cell lines","Cell lines"), Dependency_Info = c("No fusion-associated\ndependency","Other TAD gene\nis dependency","Partner\nis dependency","No fusion-associated\ndependency","Other TAD gene\nis dependency","Partner\nis dependency"), Count = c(length(fusion_with_no_associated_dependency)/length(total_fusions_with_dependency_data)*100,length(fusions_with_unique_other_tad_dependency)/length(total_fusions_with_dependency_data)*100,length(fusions_with_partner_dependency)/length(total_fusions_with_dependency_data)*100,length(cell_lines_with_no_associated_dependency)/length(total_cell_lines_with_dependency_data)*100,length(cell_lines_with_unique_other_tad_dependency)/length(total_cell_lines_with_dependency_data)*100,length(cell_lines_with_partner_dependency)/length(total_cell_lines_with_dependency_data)*100))

ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20 <- ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20 %>% mutate(Aggregated_Dependency_Info = ifelse(Dependency_Info == "No fusion-associated\ndependency", "No fusion-associated\ndependency", "Fusion-associated\ndependency"))

ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Dependency_Info <- factor(ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Dependency_Info, levels = c("No fusion-associated\ndependency","Other TAD gene\nis dependency","Partner\nis dependency"))

ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Fusion_or_Cell_Lines <- factor(ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Fusion_or_Cell_Lines, levels = c("Fusions","Cell lines"))

ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Aggregated_Dependency_Info <- factor(ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20$Aggregated_Dependency_Info, levels = c("No fusion-associated\ndependency","Fusion-associated\ndependency"))

ALL_fusions_tibble_with_tad_dependencies_summary_plot_3_1_20 <- ggplot(ALL_fusions_tibble_with_tad_dependencies_summary_3_1_20 %>% group_by(Aggregated_Dependency_Info, Fusion_or_Cell_Lines) %>% summarise(Count = sum(Count)),aes(fill = Aggregated_Dependency_Info, x = Fusion_or_Cell_Lines, y = Count))+geom_bar(color = "black", position = "stack", stat = "identity") + labs(x = "", y = "Percentage", fill = "") + scale_fill_manual(values = c("#E6E7E8","Dark blue"))  + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text = element_text(size = 32, color = "black"), axis.title = element_text(size = 32, color = "black"), title = element_text(size = 32, color = "black"), legend.text = element_text(size = 32, color = "black"), legend.position = "right", legend.box="vertical", legend.key.height = unit(1,"inches"), legend.key.width = unit(1,"inches")) + guides(fill = guide_legend(nrow = 3)) + coord_cartesian(ylim = c(0,110))+ scale_y_continuous(breaks = c(0,20,40,60,80,100))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","ALL_fusions_tibble_with_tad_dependencies_summary_plot_3_1_20.png"), plot = ALL_fusions_tibble_with_tad_dependencies_summary_plot_3_1_20, height = 12, width = 12)

#Cell lines with dependencies

cell_lines_with_partner_dependency_by_disease <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(DepMap_ID %in% cell_lines_with_partner_dependency) %>% group_by(primary_disease) %>% summarise(partner_dependency = n_distinct(DepMap_ID))

cell_lines_with_unique_other_tad_dependency_by_disease <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(DepMap_ID %in% cell_lines_with_unique_other_tad_dependency) %>% group_by(primary_disease) %>% summarise(other_tad_dependency= n_distinct(DepMap_ID))

cell_lines_with_no_associated_dependency_by_disease <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(DepMap_ID %in% cell_lines_with_no_associated_dependency) %>% group_by(primary_disease) %>% summarise(no_dependency = n_distinct(DepMap_ID))

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 <- full_join(cell_lines_with_partner_dependency_by_disease,cell_lines_with_unique_other_tad_dependency_by_disease)

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 <- full_join(ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20,cell_lines_with_no_associated_dependency_by_disease)

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 <- ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 %>% gather(key = "dependency_category", value = "count", -primary_disease)

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 <- ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 %>% replace_na(list(count = 0))

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$dependency_category <- recode(ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$dependency_category, partner_dependency = "Partner is dependency", other_tad_dependency = "Other TAD gene is dependency", no_dependency = "No fusion-associated dependency")
  
ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$dependency_category <- factor(ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$dependency_category, levels = c("No fusion-associated dependency","Other TAD gene is dependency","Partner is dependency"))

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 <- ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 %>% mutate(aggregated_dependency_category = ifelse(dependency_category == "No fusion-associated dependency", "No fusion-associated dependency", "Fusion-associated dependency"))

ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$aggregated_dependency_category <- factor(ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20$aggregated_dependency_category, levels = c("No fusion-associated dependency","Fusion-associated dependency"))

ALL_fusions_tibble_with_tad_dependencies_by_disease_plot_4_27_20 <- ggplot(ALL_fusions_tibble_with_tad_dependencies_by_disease_4_27_20 %>% group_by(aggregated_dependency_category, primary_disease) %>% summarise(count = sum(count)),aes(fill = aggregated_dependency_category, x = reorder(primary_disease, -count), y = count))+geom_bar(color = "black", position = "stack", stat = "identity") + labs(x = "", y = "Count", fill = "") + scale_fill_manual(values = c("#E6E7E8","Dark blue")) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 18, color = "black", angle = 90, hjust = 1, vjust = 0.5), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","ALL_fusions_tibble_with_tad_dependencies_by_disease_plot_4_27_20.png"), plot = ALL_fusions_tibble_with_tad_dependencies_by_disease_plot_4_27_20, height = 8, width = 10)

#Unique dependencies and fusion-dependency pairings
partner_dependencies <- distinct_fusion_dependency_pairings %>% filter(is_partner == TRUE) %>% distinct(Gene) %>% unlist() %>% unname()

unique_other_tad_dependencies <- setdiff(distinct_fusion_dependency_pairings %>% filter(is_partner == FALSE) %>% distinct(Gene) %>% unlist() %>% unname(), partner_dependencies)

fusion_dependency_pairings_with_partner_dependencies <- distinct_fusion_dependency_pairings %>% filter(is_partner == TRUE) %>% mutate(Fusion_Dependency_Pairing = str_c(Fusion,"-",Gene)) %>% distinct(Fusion_Dependency_Pairing) %>% unlist() %>% unname()

fusion_dependency_pairings_with_other_tad_dependencies <- setdiff(distinct_fusion_dependency_pairings %>% filter(is_partner == FALSE) %>% mutate(Fusion_Dependency_Pairing = str_c(Fusion,"-",Gene)) %>% distinct(Fusion_Dependency_Pairing) %>% unlist() %>% unname(), fusion_dependency_pairings_with_partner_dependencies)

fusion_dependency_pairings_with_partner_dependencies_duplicates_removed <- distinct_fusion_dependency_cell_line_pairings_duplicates_removed %>% mutate(gene_cell_line_pairing = str_c(gene,"-",cell_lines)) %>% filter(is_partner == TRUE) %>% distinct(gene_cell_line_pairing) %>% unlist() %>% unname()

fusion_dependency_pairings_with_other_tad_dependencies_duplicates_removed <- setdiff(distinct_fusion_dependency_cell_line_pairings_duplicates_removed %>% mutate(gene_cell_line_pairing = str_c(gene,"-",cell_lines)) %>% filter(is_partner == FALSE) %>% distinct(gene_cell_line_pairing) %>% unlist() %>% unname(), fusion_dependency_pairings_with_partner_dependencies_duplicates_removed)

unique_fusion_associated_dependency_summary_3_1_20 <- tibble(summary_type = c("Dependencies", "Dependencies", "Fusion-Dependency Pairings", "Fusion-Dependency Pairings", "Fusion-Dependency Pairings (Duplicates Removed)","Fusion-Dependency Pairings (Duplicates Removed)"), dependency_type = c("Partner dependency", "Collateral dependency", "Partner dependency", "Collateral dependency", "Partner dependency", "Collateral dependency"), count = c(length(partner_dependencies),length(unique_other_tad_dependencies),length(fusion_dependency_pairings_with_partner_dependencies),length(fusion_dependency_pairings_with_other_tad_dependencies),length(fusion_dependency_pairings_with_partner_dependencies_duplicates_removed),length(fusion_dependency_pairings_with_other_tad_dependencies_duplicates_removed)))

unique_fusion_associated_dependency_summary_3_1_20$summary_type <- factor(unique_fusion_associated_dependency_summary_3_1_20$summary_type, levels = c("Dependencies", "Fusion-Dependency Pairings", "Fusion-Dependency Pairings (Duplicates Removed)"))

unique_fusion_associated_dependency_summary_plot_3_1_20 <- ggplot(unique_fusion_associated_dependency_summary_3_1_20 %>% filter(!summary_type == "Dependencies"), aes(fill = dependency_type, x = summary_type, y = count))+geom_bar(position = "stack", stat = "identity", color = "black") + labs(x = "", y = "Count", fill = "") + scale_fill_manual(labels = c("Collateral","Partner"), values = c("#B9DBF4","#317EC2")) + scale_x_discrete(labels = function(x) str_wrap(x, width = 15)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 18, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.text = element_text(size = 20, color = "black"), legend.position = "right",legend.box="vertical", legend.margin=margin())+ guides(fill = guide_legend(nrow = 2))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","unique_fusion_associated_dependency_summary_plot_3_1_20.png"), plot = unique_fusion_associated_dependency_summary_plot_3_1_20, height = 7, width = 6)

#Average number of high-confidence fusions per cell line by disease type
mean_fusions_per_cell_line_by_disease <- ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% group_by(primary_disease) %>% summarise(total_fusions = n_distinct(Fusion), total_cell_lines = n_distinct(stripped_cell_line_name)) %>% mutate(mean_fusions_per_cell_line = total_fusions/total_cell_lines)

mean_fusions_per_cell_line_by_disease_plot <- ggplot(mean_fusions_per_cell_line_by_disease,aes(fill = "", x = reorder(primary_disease, -mean_fusions_per_cell_line), y = mean_fusions_per_cell_line))+geom_bar(color = "black", position = "stack", stat = "identity") + labs(x = "", y = "Mean fusion count / cell line", fill = "") + scale_fill_manual(values = c("Dark blue")) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 18, color = "black", angle = 90, hjust = 1, vjust = 0.5), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","mean_fusions_per_cell_line_by_disease_plot.png"), plot = mean_fusions_per_cell_line_by_disease_plot, height = 8, width = 10)

#The distributions of random sampling from null empirical distributions

random_sampling_from_null_empirical_distributions <- tibble(iteration = c(1:1000), multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles = multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles, multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles = multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles)

multiple_iterations_cumulative_partner_dependencies_plot_3_1_20 <- ggplot(random_sampling_from_null_empirical_distributions, aes(x = multiple_iterations_cumulative_partner_dependencies_permuted_by_expression_quartiles)) + geom_density(size = 1, fill = "#E6E7E8") + geom_segment(data = count_of_actual_fusion_dependency_pairings_conservative %>% filter(partner_or_other_TAD == "partner"), aes(x = count, xend = count, y = .05, yend = 0, size = 2, color = partner_or_other_TAD), arrow = arrow(length = unit(0.5,"cm"))) +scale_color_manual(values = c("#317EC2")) + labs(x = "Partner fusion-dependency pairings", y = "Density") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank(), axis.text.x = element_text(size = 24, color = "black"), axis.title = element_text(size = 24, color = "black"), title = element_text(size = 24, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","multiple_iterations_cumulative_partner_dependencies_plot_3_1_20.png"), plot = multiple_iterations_cumulative_partner_dependencies_plot_3_1_20)

multiple_iterations_cumulative_other_TAD_dependencies_plot_3_1_20 <- ggplot(random_sampling_from_null_empirical_distributions, aes(x = multiple_iterations_cumulative_other_TAD_dependencies_permuted_by_expression_quartiles)) + geom_density(size = 1, fill = "#E6E7E8") + geom_segment(data = count_of_actual_fusion_dependency_pairings_conservative %>% filter(partner_or_other_TAD == "other_TAD"), aes(x = count, xend = count, y = .02, yend = 0, size = 2, color = partner_or_other_TAD), arrow = arrow(length = unit(0.5,"cm"))) +scale_color_manual(values = c("#B9DBF4")) + labs(x = "Collateral fusion-dependency pairings", y = "Density") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank(), axis.text.x = element_text(size = 24, color = "black"), axis.title = element_text(size = 24, color = "black"), title = element_text(size = 24, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","multiple_iterations_cumulative_other_TAD_dependencies_plot_3_1_20.png"), plot = multiple_iterations_cumulative_other_TAD_dependencies_plot_3_1_20)

#The distributions of random sampling from null empirical distributions (phenotypic_permutation)

random_sampling_from_null_empirical_distributions_phenotypic_permutation <- tibble(iteration = c(1:1000), count_of_partner_dependencies_phenotypic_permutation = count_of_partner_dependencies_phenotypic_permutation, count_of_other_TAD_dependencies_phenotypic_permutation = count_of_other_TAD_dependencies_phenotypic_permutation)

multiple_iterations_cumulative_partner_dependencies_plot_phenotypic_permutation <- ggplot(random_sampling_from_null_empirical_distributions_phenotypic_permutation, aes(x = count_of_partner_dependencies_phenotypic_permutation)) + geom_density(size = 1, fill = "#E6E7E8") + geom_segment(data = count_of_actual_fusion_dependency_pairings_conservative %>% filter(partner_or_other_TAD == "partner"), aes(x = count, xend = count, y = .05, yend = 0, size = 2, color = partner_or_other_TAD), arrow = arrow(length = unit(0.5,"cm"))) +scale_color_manual(values = c("#317EC2")) + labs(x = "Partner fusion-dependency pairings", y = "Density") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank(), axis.text.x = element_text(size = 24, color = "black"), axis.title = element_text(size = 24, color = "black"), title = element_text(size = 24, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","multiple_iterations_cumulative_partner_dependencies_plot_phenotypic_permutation.png"), plot = multiple_iterations_cumulative_partner_dependencies_plot_3_1_20)

ggplot(random_sampling_from_null_empirical_distributions_phenotypic_permutation, aes(x = count_of_other_TAD_dependencies_phenotypic_permutation)) + geom_density(size = 1, fill = "#E6E7E8") + geom_segment(data = count_of_actual_fusion_dependency_pairings_conservative %>% filter(partner_or_other_TAD == "other_TAD"), aes(x = count, xend = count, y = .02, yend = 0, size = 2, color = partner_or_other_TAD), arrow = arrow(length = unit(0.5,"cm"))) +scale_color_manual(values = c("#B9DBF4")) + labs(x = "Collateral fusion-dependency pairings", y = "Density") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank(), axis.text.x = element_text(size = 24, color = "black"), axis.title = element_text(size = 24, color = "black"), title = element_text(size = 24, color = "black"), legend.position = "none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","multiple_iterations_cumulative_other_TAD_dependencies_plot_phenotypic_permutation.png"), plot = multiple_iterations_cumulative_other_TAD_dependencies_plot_3_1_20)

#Overlap of results from genome-scale approach to identifying fusion-dependency pairings with fusion-associated dependency t-statistic analysis
venn.diagram(x = list(fusion_associated_dependencies_with_FDRs_TSTATS$Fusion_Dependency_Pairing,fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.05_duplicates_removed$Fusion_Dependency_Pairing), category.names = c("Fusion-dependency pairings based on\ngenome-scale t-tests with BH correction (FDR < .05)","Fusion-dependency pairings based on\npartner TAD gene t-tests only with permutation-based\ncorrection of t-statistics (FDR < .05)"), filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","fusion_dependency_pairing_for_different_methods_of_FDR_permutation_FDR_point05_venn_diagram_6_22_20.png"), output = TRUE, height = 8, width = 12, units = "in", main = "", fill = c("Dark Blue", "#080808"), alpha = .5, main.cex = 4, main.fontfamily = "sans", cex = 0, fontface = "bold", fontfamily = "sans", cat.dist = c(0.05, 0.05), cat.just = list(c(1,-13),c(0,-10)), cat.default.pos = "outer", cat.cex = 0, cat.fontface = "bold", cat.fontfamily = "sans")

venn.diagram(x = list(fusion_associated_dependencies_with_FDRs_TSTATS$Fusion_Dependency_Pairing,fusion_associated_dependency_TSTAT_FDR_NA_removed_FDR_0.1_duplicates_removed$Fusion_Dependency_Pairing), category.names = c("Fusion-dependency pairings based on\ngenome-scale t-tests with BH correction (FDR < .05)","Fusion-dependency pairings based on\npartner TAD gene t-tests only with permutation-based\ncorrection of t-statistics (FDR < .1)"), filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","fusion_dependency_pairing_for_different_methods_of_FDR_permutation_FDR_point1_venn_diagram_6_22_20.png"), output = TRUE, height = 8, width = 12, units = "in", main = "", fill = c("Dark Blue", "#080808"), alpha = .5, main.cex = 4, main.fontfamily = "sans", cex = 0, fontface = "bold", fontfamily = "sans", cat.dist = c(0.05, 0.05), cat.just = list(c(1,-14),c(0,-10)), cat.default.pos = "outer", cat.cex = 0, cat.fontface = "bold", cat.fontfamily = "sans", ext.line.lwd	= 0)

#Generating dependency and expression volcano plots for fusions with associated dependencies
counter = 0

for(fusion in unique(distinct_fusion_dependency_pairings$Fusion)){

ggplot(dependency_probability_fusion_with_vs_without_stats[[fusion]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1)  + geom_point(data=dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% c(all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]], fusion_partners[[fusion]]) & dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% c(all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]], fusion_partners[[fusion]]) & dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "red", size = 3) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = str_c("Fusion: ", fusion)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(),  axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)
  
ggsave(filename = here::here("results/fusion_dependency_probability_plots_for_associated_dependencies",str_c(fusion,".png")))

ggplot(expression_fusion_with_vs_without_stats[[fusion]], aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1)  + geom_point(data=expression_fusion_with_vs_without_stats[[fusion]] %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=expression_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% c(all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]], fusion_partners[[fusion]]) & log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + geom_point(data=expression_fusion_with_vs_without_stats[[fusion]] %>% filter(gene %in% c(all_unique_coding_TAD_neighbors_with_dependency_data[[fusion]], fusion_partners[[fusion]]) & log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "red", size = 3) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = "Pan-cancer differential expression:", subtitle = fusion) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none") 

ggsave(filename = here::here("results/fusion_RNA_expression_plots_for_associated_dependencies",str_c(fusion,".png")))

counter = counter + 1
print(counter)

}

#Specific dependency examples

#EWSR1-ERG
ggplot(dependency_probability_fusion_with_vs_without_stats[["EWSR1-ERG"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["EWSR1-ERG"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["EWSR1-ERG"]] %>% filter(gene %in% fusion_partners[["EWSR1-ERG"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: EWSR1-ERG") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","EWSR1-ERG_dependency.png"))

#EWSR1-FLI1
ggplot(dependency_probability_fusion_with_vs_without_stats[["EWSR1-FLI1"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["EWSR1-FLI1"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["EWSR1-FLI1"]] %>% filter(gene %in% fusion_partners[["EWSR1-FLI1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["EWSR1-FLI1"]] %>% filter(gene == "FLI1"), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "red", size = 3)  + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: EWSR1-FLI1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","EWSR1-FLI1_dependency.png"))

#IGH-NSD2
ggplot(dependency_probability_fusion_with_vs_without_stats[["IGH-NSD2"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["IGH-NSD2"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["IGH-NSD2"]] %>% filter(gene %in% c("NSD2","FGFR3")), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["IGH-NSD2"]] %>% filter(gene == "FGFR3"), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "red", size = 3)  + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: IGH-NSD2") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","IGH-NSD2_dependency.png"))

#BCR-ABL1
ggplot(dependency_probability_fusion_with_vs_without_stats[["BCR-ABL1"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["BCR-ABL1"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["BCR-ABL1"]] %>% filter(gene %in% fusion_partners[["BCR-ABL1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["BCR-ABL1"]] %>% filter(gene %in% fusion_partners[["BCR-ABL1"]]), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "red", size = 3) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: BCR-ABL1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","BCR-ABL1_dependency.png"))

#Generic mock example
ggplot(dependency_probability_fusion_with_vs_without_stats[["TMPRSS2-ERG"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3", size = 3) + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["TMPRSS2-ERG"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black", size = 3) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["TMPRSS2-ERG"]] %>% filter(gene == "ERG"), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "red", size = 7) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = str_c("Dependency space for\nspecific fusion")) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 24, color = "black"), axis.text.x = element_text(size = 24, color = "black"), axis.title = element_text(size = 24, color = "black"), title = element_text(size = 24, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","mock_dependency.png"))

#TPM3-NTRK1
ggplot(dependency_probability_fusion_with_vs_without_stats[["TPM3-NTRK1"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["TPM3-NTRK1"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["TPM3-NTRK1"]] %>% filter(gene %in% fusion_partners[["TPM3-NTRK1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: TPM3-NTRK1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","TPM3-NTRK1_dependency.png"))

#TCF3-PBX1
ggplot(dependency_probability_fusion_with_vs_without_stats[["TCF3-PBX1"]], aes(x = dependency_probability_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 0.5, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -0.5, color = "blue", linetype = "dashed", size = 1) + geom_point(data=dependency_probability_fusion_with_vs_without_stats[["TCF3-PBX1"]] %>% filter(dependency_probability_difference > 0.5 & negative_log10_BH_QValue > -log10(.05)), aes(x = dependency_probability_difference, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=dependency_probability_fusion_with_vs_without_stats[["TCF3-PBX1"]] %>% filter(gene %in% fusion_partners[["TCF3-PBX1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2) + labs(x = 'Difference in dependency probability', y = expression(-log[10](Q~value)), title = "Fusion: TCF3-PBX1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + xlim(-1,1)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","TCF3-PBX1_dependency.png"))

#Specific expression examples

#BCR-ABL1
ggplot(expression_fusion_with_vs_without_stats[["BCR-ABL1"]], aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_point(data=expression_fusion_with_vs_without_stats[["BCR-ABL1"]] %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=expression_fusion_with_vs_without_stats[["BCR-ABL1"]] %>% filter(gene %in% fusion_partners[["BCR-ABL1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2, nudge_x = -2) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = "Pan-cancer differential expression:", subtitle = "BCR-ABL1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","BCR-ABL1_expression.png"))

#KMT2A-AFDN
ggplot(expression_fusion_with_vs_without_stats[["KMT2A-AFDN"]], aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_point(data=expression_fusion_with_vs_without_stats[["KMT2A-AFDN"]] %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=expression_fusion_with_vs_without_stats[["KMT2A-AFDN"]] %>% filter(gene %in% fusion_partners[["KMT2A-AFDN"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2, nudge_x = -2) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = "Pan-cancer differential expression:", subtitle = "KMT2A-AFDN") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","KMT2A-AFDN_expression.png"))

#KMT2A-AFF1
ggplot(expression_fusion_with_vs_without_stats[["KMT2A-AFF1"]], aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_point(data=expression_fusion_with_vs_without_stats[["KMT2A-AFF1"]] %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=expression_fusion_with_vs_without_stats[["KMT2A-AFF1"]] %>% filter(gene %in% fusion_partners[["KMT2A-AFF1"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2, nudge_x = -2) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = "Pan-cancer differential expression:", subtitle = "KMT2A-AFF1") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","KMT2A-AFF1_expression.png"))

#KMT2A-MLLT3
ggplot(expression_fusion_with_vs_without_stats[["KMT2A-MLLT3"]], aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_point(data=expression_fusion_with_vs_without_stats[["KMT2A-MLLT3"]] %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=expression_fusion_with_vs_without_stats[["KMT2A-MLLT3"]] %>% filter(gene %in% fusion_partners[["KMT2A-MLLT3"]]), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2, nudge_x = -2) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = "Pan-cancer differential expression:", subtitle = "KMT2A-MLLT3") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), plot.subtitle = element_text(size = 20, color = "black"), legend.position="none")

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","KMT2A-MLLT3_expression.png"))

#Supporting WGS data for fusions
list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary$evidence_of_SV_in_WGS_data <- factor(list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary$evidence_of_SV_in_WGS_data, levels = c("No WGS evidence","Right gene involved in SV","Left gene involved in SV","Both genes involved in SV","Fusion seen"))

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary <- list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary %>% mutate(evidence_of_SV_in_WGS_data_aggregated = ifelse(evidence_of_SV_in_WGS_data == "No WGS evidence","No WGS evidence",ifelse(evidence_of_SV_in_WGS_data == "Fusion seen","Exact fusion seen","At least one gene in SV")))

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary$evidence_of_SV_in_WGS_data_aggregated <- factor(list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary$evidence_of_SV_in_WGS_data_aggregated, levels = c("No WGS evidence","At least one gene in SV","Exact fusion seen"))

list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary_plot_3_1_20 <- ggplot(list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary %>% group_by(evidence_of_SV_in_WGS_data_aggregated) %>% summarise(percentage = sum(percentage)), aes(fill = evidence_of_SV_in_WGS_data_aggregated, x = NA, y = percentage))+geom_bar(position = "stack", stat = "identity", color = "black") + labs(x = "", y = "Percentage", fill = "") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_blank(), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_blank(), axis.text.x = element_blank(), axis.title = element_blank(), title = element_text(size = 24, color = "black"), legend.text = element_text(size = 24, color = "black"), legend.position="none", legend.box="vertical", legend.margin=margin()) + scale_fill_manual(values = c("#E6E7E8","Light green","Dark green")) + guides(fill = guide_legend(nrow = 5))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary_plot_3_1_20.png"), plot = list_of_fusions_with_related_dependencies_and_associated_cell_lines_with_orthogonal_WGS_data_combined_summary_plot_3_1_20, height = 7, width = 2)

#Evidence for fusions or partners in TCGA dataset
summary_of_tcga_samples_fusions_with_tcga_disease_rep <- summary_of_tcga_samples_fusions_with_tcga_disease_rep %>% gather(key = "fusion_or_partner", value = "percent", 2:3)

summary_of_tcga_samples_fusions_with_tcga_disease_rep$fusion_or_partner <- summary_of_tcga_samples_fusions_with_tcga_disease_rep$fusion_or_partner %>% recode(fusion_percent = "Exact\nfusion", partner_percent = "One\nfusion\npartner")

summary_of_tcga_samples_fusions_with_tcga_disease_rep$frequency <- factor(summary_of_tcga_samples_fusions_with_tcga_disease_rep$frequency, levels = c("0","1","2 - 5","5+"))

summary_of_tcga_samples_fusions_with_tcga_disease_rep <- summary_of_tcga_samples_fusions_with_tcga_disease_rep %>% mutate(aggregated_frequency = ifelse(frequency == "0","0",ifelse(frequency == "5+","5+","1 - 5")))

summary_of_tcga_samples_fusions_with_tcga_disease_rep$aggregated_frequency <- factor(summary_of_tcga_samples_fusions_with_tcga_disease_rep$aggregated_frequency, levels = c("0","1 - 5","5+"))

summary_of_tcga_samples_plot <- ggplot(summary_of_tcga_samples_fusions_with_tcga_disease_rep %>% group_by(aggregated_frequency,fusion_or_partner) %>% summarise(percent = sum(percent)), aes(fill = aggregated_frequency, x = fusion_or_partner, y = percent))+geom_bar(position = "stack", stat = "identity", color = "black") + labs(fill = "Count in TCGA", x = "", y = "Percentage", fill = "") + scale_fill_brewer(palette = "Greens") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), legend.title = element_text(size = 20, color = "black"), legend.text = element_text(size = 20, color = "black"), legend.position="right", legend.box="vertical", legend.margin=margin()) + scale_fill_manual(values = c("#E6E7E8","Light green","Dark green")) + coord_cartesian(ylim = c(0,100))+ scale_y_continuous(breaks = c(0,20,40,60,80,100))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","summary_of_tcga_samples_plot.png"), plot = summary_of_tcga_samples_plot, height = 7, width = 6)

#Venn diagrams of partner-gene pairings that are dependencies and those that are overexpressed

fusion_overexpression_pairings_with_partners <- TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed %>% filter(is_partner == TRUE) %>% mutate(Fusion_Overexpression_Pairing = str_c(Fusion,"-",Gene)) %>% distinct(Fusion_Overexpression_Pairing) %>% unlist() %>% unname()

fusion_overexpression_pairings_with_other_tad_genes <- setdiff(TAD_signficant_expression_fusions_and_cell_lines_ALL_with_boundaries_duplicates_removed %>% filter(is_partner == FALSE) %>% mutate(Fusion_Overexpression_Pairing = str_c(Fusion,"-",Gene)) %>% distinct(Fusion_Overexpression_Pairing) %>% unlist() %>% unname(), fusion_overexpression_pairings_with_partners)

partner_dependencies_and_overexpressed_genes_venn_diagram_3_1_20 <- venn.diagram(x = list(fusion_dependency_pairings_with_partner_dependencies,fusion_overexpression_pairings_with_partners), category.names = c("",""), filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","partner_dependencies_and_overexpressed_genes_venn_diagram_3_1_20.png"), output = TRUE, height = 2, width = 3, units = "in", main = "", fill = c("#317EC2", "#080808"), alpha = .5, main.cex = 4, main.fontfamily = "arial", cex = 0, fontface = "bold", fontfamily = "sans", cat.pos = 0, cat.default.pos = "outer", cat.cex = 1, cat.fontface = "bold", cat.fontfamily = "arial")

other_TAD_dependencies_and_overexpressed_genes_venn_diagram_3_1_20 <- venn.diagram(x = list(fusion_dependency_pairings_with_other_tad_dependencies,fusion_overexpression_pairings_with_other_tad_genes), category.names = c("",""), filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","other_TAD_dependencies_and_overexpressed_genes_venn_diagram_3_1_20.png"), output = TRUE, height = 2, width = 3, units = "in", main = "", fill = c("#B9DBF4", "#080808"), alpha = .5, main.cex = 4,  main.fontfamily = "arial", cex = 0, ext.dist = -0.5, ext.line.lwd	= 0, fontface = "bold", fontfamily = "sans", cat.pos = 0, cat.default.pos = "outer", cat.cex = 1, cat.fontface = "bold", cat.fontfamily = "sans")

#Summary copy number data
copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral <- recode(copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral, `no data` = "No data", neutral = "Neutral", loss = "Loss", gain = "Gain")

copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral <- factor(copy_number_alterations_involving_fusion_associated_dependencies_summary$loss_gain_neutral, levels = c("No data","Neutral","Loss","Gain"))

copy_number_alterations_involving_fusion_associated_dependencies_summary$dependency_type <- factor(copy_number_alterations_involving_fusion_associated_dependencies_summary$dependency_type, levels = c("Partner\ndependency","Collateral\ndependency"))

copy_number_alterations_involving_fusion_associated_dependencies_summary_plot_3_4_20 <- ggplot(copy_number_alterations_involving_fusion_associated_dependencies_summary %>% filter(!loss_gain_neutral == "No data"), aes(fill = loss_gain_neutral, x = dependency_type, y = count))+geom_bar(color = "black", position = "stack", stat = "identity") + labs(title = "Count of fusion-dependency-cell\ncontexts with copy number\nalterations affecting dependency", x = "", y = "Count", fill = "Copy number alteration") + scale_fill_manual(values = c("#E6E7E8", "Red", "Dark Blue")) + scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.text = element_text(size = 20, color = "black"), legend.position="right", legend.box="vertical", legend.margin=margin(), legend.title = element_blank())

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","copy_number_alterations_involving_fusion_associated_dependencies_summary_plot_3_4_20.png"), plot = copy_number_alterations_involving_fusion_associated_dependencies_summary_plot_3_4_20, height = 8, width = 6)

#Summary mutation data

mutations_involving_fusion_associated_dependencies_summary <- mutations_involving_fusion_associated_dependencies_summary %>% mutate(mutation_in_dependency = ifelse(variant_in_dependency == TRUE, "Mutation", "No mutation"))

mutations_involving_fusion_associated_dependencies_summary$mutation_in_dependency <- factor(mutations_involving_fusion_associated_dependencies_summary$mutation_in_dependency, levels = c("No mutation", "Mutation"))

mutations_involving_fusion_associated_dependencies_summary$dependency_type <- factor(mutations_involving_fusion_associated_dependencies_summary$dependency_type, levels = c("Partner\ndependency","Collateral\ndependency"))

mutations_involving_fusion_associated_dependencies_summary <- mutations_involving_fusion_associated_dependencies_summary %>% remove_missing()

mutations_involving_fusion_associated_dependencies_summary_plot_3_4_20 <- ggplot(mutations_involving_fusion_associated_dependencies_summary, aes(fill = mutation_in_dependency, x = dependency_type, y = count))+geom_bar(color = "black", position = "stack", stat = "identity") + labs(title = "Count of fusion-dependency-cell\ncontexts with mutations\naffecting dependency", x = "", y = "Count", fill = "Mutation in dependency?") + scale_fill_manual(labels = c("None","Mutation"), values = c("#E6E7E8","Dark Blue"))  + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.text = element_text(size = 20, color = "black"), legend.position="right", legend.box="vertical", legend.margin=margin(), legend.title = element_blank())

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","mutations_involving_fusion_associated_dependencies_summary_plot_3_4_20.png"), plot = mutations_involving_fusion_associated_dependencies_summary_plot_3_4_20, height = 8, width = 6)

#Pie chart of proportion of Cosmic fusions with dependency data that have an associated dependency

ALL_fusions_protein_coding_filtered_list_final_with_dependency_data %>% filter(DepMap_ID %in% colnames(ccle_dependency_probability_matrix) & Cosmic == TRUE) %>% summarise(n_distinct(Fusion)) %>% view()

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% filter(cosmic_fusion == TRUE & is_partner == TRUE) %>% summarise(n_distinct(fusion)) %>% view()

RNA_expression_copy_number_and_mutation_data_by_dependency_fusion_and_cell_line %>% filter(cosmic_fusion == TRUE & is_partner == FALSE) %>% summarise(n_distinct(fusion)) %>% view()

pie_chart_proportion_of_cosmic_fusions_with_associated_dependency <- pie(x = c(19,1,19), labels = NA, clockwise = FALSE, col = c("#317EC2","#B9DBF4", "#E6E7E8"))

#Bar plots of COSMIC census genes and kinases
percentage_of_fusion_where_associated_dependency_is_cosmic_census_gene <- ggplot(fusion_dependency_co_descriptor_plot_cosmic_annotation_summary, aes(x = cosmic_fusion, y = dependency_cosmic_census_gene_percent))+geom_bar(color = "black", fill = "dark blue", stat = "identity") + labs(title = "% of fusions with dependency\nas COSMIC cancer census gene", x = "", y = "") + scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none") + scale_y_continuous(limits = c(0,100), breaks = c(0,20,40,60,80,100))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","percentage_of_fusion_where_associated_dependency_is_cosmic_census_gene.png") , plot = percentage_of_fusion_where_associated_dependency_is_cosmic_census_gene, height = 8, width = 6)

percentage_of_fusion_where_associated_dependency_is_kinase <- ggplot(fusion_dependency_co_descriptor_plot_cosmic_annotation_summary, aes(x = cosmic_fusion, y = dependency_kinase_percent))+geom_bar(color = "black", fill = "dark blue", stat = "identity") + labs(title = "% of fusions with dependency\nas kinase gene", x = "", y = "") + scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none") + scale_y_continuous(limits = c(0,25), breaks = c(0,5,10,15,20,25))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","percentage_of_fusion_where_associated_dependency_is_kinase.png") , plot = percentage_of_fusion_where_associated_dependency_is_kinase, height = 8, width = 6)

percentage_of_fusion_where_fusion_partner_is_cosmic_census_gene <- ggplot(fusion_dependency_co_descriptor_plot_cosmic_annotation_summary, aes(x = cosmic_fusion, y = fusion_partner_cosmic_census_gene_percent))+geom_bar(color = "black", fill = "dark blue", stat = "identity") + labs(title = "% of fusions with partner\nas COSMIC cancer census gene", x = "", y = "") + scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none") + scale_y_continuous(limits = c(0,100), breaks = c(0,20,40,60,80,100))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","percentage_of_fusion_where_fusion_partner_is_cosmic_census_gene.png") , plot = percentage_of_fusion_where_fusion_partner_is_cosmic_census_gene, height = 8, width = 6)

percentage_of_fusion_where_fusion_partner_is_kinase <- ggplot(fusion_dependency_co_descriptor_plot_cosmic_annotation_summary, aes(x = cosmic_fusion, y = fusion_partner_kinase_percent))+geom_bar(color = "black", fill = "dark blue", stat = "identity") + labs(title = "% of fusions with partner\nas kinase gene", x = "", y = "") + scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.ticks.y=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position = "none") + scale_y_continuous(limits = c(0,25), breaks = c(0,5,10,15,20,25))

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","percentage_of_fusion_where_fusion_partner_is_kinase.png") , plot = percentage_of_fusion_where_fusion_partner_is_kinase, height = 8, width = 6)


#Venetoclax drug data for B-ALL cell line with IgH-NSD2 fusion vs. all other B-ALL cell lines
#BCL2 appears as dependency in 4/6 B-ALL cell lines that also have dependency data: SEM, HB1119, SEMK2, 697, REH(weak), NALM6(weak)
venetoclax_screen_all_cell_lines <- venetoclax_screen_all_cell_lines %>% mutate(B_ALL = ifelse(DepMap_ID %in% B_ALL_cell_lines,"B-ALL cells\n(n=10)", "Other cancer cells\n(n=460)"))

ggplot(venetoclax_screen_all_cell_lines, aes(x = B_ALL, y = venetoclax_AUC)) + geom_violin(size = 2, aes(fill = B_ALL)) + scale_color_manual(values = c("Black","Black")) + scale_fill_manual(values = c("Light blue","#E6E7E8")) + geom_point(venetoclax_screen_all_cell_lines %>% filter(DepMap_ID ==	"ACH-000151"), mapping = aes(x = B_ALL, y = venetoclax_AUC), color = "red", size = 3) + geom_label_repel(venetoclax_screen_all_cell_lines %>% filter(DepMap_ID ==	"ACH-000151"), mapping = aes(x = B_ALL, y = venetoclax_AUC), label = "Cell line with\nBCL2-IGH fusion", box.padding = 2, size = 6, nudge_x = 0.5, nudge_y = 5) + labs(x = '', y = 'AUC', title = "Cell line sensitivity to venetoclax") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.text = element_text(size = 20, color = "black"), legend.position="none") + coord_cartesian(ylim = c(0,20)) + stat_compare_means(data = venetoclax_screen_all_cell_lines, aes(x = B_ALL), method = "t.test", comparisons = list(c("B-ALL cells\n(n=10)", "Other cancer cells\n(n=460)")), size = 6)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","Cell lines sensitivity to venetoclax.png"))

#Volcano plot comparing AUC drug data between multiple myeloma cells
ggplot(ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats_annotated, aes(x = auc_difference, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept =1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_label_repel(data=ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats_annotated %>% filter(cpd_name %in% c("AZD4547","nintedanib","cediranib","lenvatinib")), aes(label = cpd_name), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 1)  + geom_point(data=ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats_annotated %>% filter(negative_log10_BH_QValue > -log10(.05)), aes(x = auc_difference, y = negative_log10_BH_QValue), color = "black") + geom_point(data=ctd2_auc_myeloma_cells_IGH_NSD2_with_vs_without_stats_annotated %>% filter(cpd_name %in% c("AZD4547","nintedanib")), aes(x = auc_difference, y = negative_log10_BH_QValue), color = "red", size = 3) + labs(x = 'Difference in AUC', y = expression(-log[10](Q~value)), title = "Compound screening for multiple\nmyeloma cell lines stratified by\npresence/absence of IGH-NSD2 fusion") + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none")+ylim(0,2)+xlim(-5,5)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","Difference_in_AUC_for_multiple_myeloma_cell_lines_based_on_IGH_NSD2.png"))

#Volcano plot comparing RNA expression between multiple myeloma cells stratified by presence of IgH-NSD2 fusion
ggplot(RNA_expression_myeloma_cells_IGH_NSD2_with_vs_without_stats, aes(x = log2_fold_change, y = negative_log10_BH_QValue)) + geom_point(color = "#B5B3B3") + geom_hline(yintercept = -log10(.05), color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = 1, color = "blue", linetype = "dashed", size = 1) + geom_vline(xintercept = -1, color = "blue", linetype = "dashed", size = 1) + geom_point(data=RNA_expression_myeloma_cells_IGH_NSD2_with_vs_without_stats %>% filter(log2_fold_change > 1 & negative_log10_BH_QValue > -log10(.05)), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "black") + geom_label_repel(data=RNA_expression_myeloma_cells_IGH_NSD2_with_vs_without_stats %>% filter(gene %in% c("FGFR3","NSD2")), aes(label = gene), label.size = NA, size = 8, fontface = "bold.italic", box.padding = 2, nudge_x = 2)  + geom_point(data=RNA_expression_myeloma_cells_IGH_NSD2_with_vs_without_stats %>% filter(gene %in% c("FGFR3","NSD2")), aes(x = log2_fold_change, y = negative_log10_BH_QValue), color = "red", size = 3) + labs(x = 'Log2 Fold Change in TPM', y = expression(-log[10](Q~value)), title = str_c("Differential expression for multiple\nmyeloma cell lines stratified by presence/\nabsence of IGH-NSD2 fusion")) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none")+xlim(-9,9)

ggsave(filename = here::here("results/TAD_fusion_dependency_and_expression_summary_plots","Difference_in_RNA_expression_for_multiple_myeloma_cell_lines_based_on_IGH_NSD2.png"))

#CRISPR phenotypic kill scores of EML4 in spheroids (individual and aggregated)

ttest_comparisons_individual <- list(c("NCIH23","NCIH1975"),c("NCIH23","NCIH2009"))

ttest_comparisons_aggregate <- list(c("Fusion", "No fusion"))

phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_EML4 <- ggplot(combined_sgrna_counts_pZ_gather_EML4, aes(x = cell_line, y = pZ, color = cell_line)) + geom_hline(yintercept = 0, linetype = "dashed", color = "black", size = 1) + annotate("text", label = "Median kill score\nfor non-targeting\nguides", x = 4, y = 2, size = 5) + geom_violin(size = 2) + scale_color_manual(values = c("Red","Black","Black")) + stat_summary(fun.y=mean, geom="point", size=2, color=c("red","black","black")) + labs(title = "Phenotypic kill scores in CRISPR\nspheroid models of lung cancer cell\nlines with targeting of EML4", x = "", y = "")+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + coord_cartesian(ylim = c(-7,5)) + scale_x_discrete(labels = c("NCIH23\n(Fusion\npresent)","NCIH1975\n(No fusion)","NCIH2009\n(No fusion)",""), limits = c("NCIH23","NCIH1975","NCIH2009","")) + stat_compare_means(method = "t.test", comparisons = ttest_comparisons_individual, size = 6, vjust = -1, bracket.size = 0)

ggsave(filename = here::here("results/organoid_crispr_plots","phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_EML4.png"), plot = phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_EML4)

phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_EML4 <- ggplot(combined_sgrna_counts_pZ_gather_EML4, aes(x = cell_line_with_or_without_fusion, y = pZ, color = cell_line_with_or_without_fusion)) + geom_hline(yintercept = 0, linetype = "dashed", color = "black", size = 1) + annotate("text", label = "Median kill score\nfor non-targeting\nguides", x = 3, y = 2, size = 7) + geom_violin(size = 2) + scale_color_manual(values = c("Red","Black")) + stat_summary(fun.y=mean, geom="point", size=2, color=c("red","black")) + labs(title = "Phenotypic kill scores in CRISPR\nspheroid models of lung cancer cell\nlines with targeting of EML4", x = "", y = "")+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + coord_cartesian(ylim = c(-7,5)) + scale_x_discrete(labels = c("NCIH23\n(Fusion present)","NCIH1975,\nNCIH2009\n(No fusion)",""), limits = c("Fusion","No fusion","")) + stat_compare_means(method = "t.test", comparisons = ttest_comparisons_aggregate, size = 6, vjust = -1, bracket.size = 0)

ggsave(filename = here::here("results/organoid_crispr_plots","phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_EML4.png"), plot = phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_EML4)

#CRISPR MEAN phenotypic kill scores of nonessential genes in spheroids (individual and aggregated)

mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_all_nonessential <- ggplot(mean_pZ_nonessentials, aes(x = cell_line, y = mean_pZ, color = cell_line)) + geom_hline(yintercept = 0, linetype = "dashed", color = "black", size = 1) + annotate("text", label = "Median kill score\nfor non-targeting\nguides", x = 4, y = 2, size = 5) + geom_jitter () + scale_color_manual(values = c("Red","Black","Black")) + labs(title = "Mean phenotypic kill scores in CRISPR\nspheroid models of lung cancer cell\nlines with targeting of nonessentials", x = "", y = "")+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + coord_cartesian(ylim = c(-7,5)) + scale_x_discrete(labels = c("NCIH23\n(Fusion\npresent)","NCIH1975\n(No fusion)","NCIH2009\n(No fusion)",""), limits = c("NCIH23","NCIH1975","NCIH2009",""))

ggsave(filename = here::here("results/organoid_crispr_plots","mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_all_nonessential.png"), plot = mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_all_nonessential)

mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_all_nonessential <- ggplot(mean_pZ_nonessentials, aes(x = cell_line_with_or_without_fusion, y = mean_pZ, color = cell_line_with_or_without_fusion)) + geom_hline(yintercept = 0, linetype = "dashed", color = "black", size = 1) + annotate("text", label = "Median kill score\nfor non-targeting\nguides", x = 3, y = 2, size = 7) + geom_jitter () + scale_color_manual(values = c("Red","Black")) + labs(title = "Mean phenotypic kill scores in CRISPR\nspheroid models of lung cancer cell\nlines with targeting of nonessentials", x = "", y = "")+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), axis.ticks.x=element_blank(), axis.text.y = element_text(size = 20, color = "black"), axis.text.x = element_text(size = 20, color = "black"), axis.title = element_text(size = 20, color = "black"), title = element_text(size = 20, color = "black"), legend.position="none") + coord_cartesian(ylim = c(-7,5)) + scale_x_discrete(labels = c("NCIH23\n(Fusion present)","NCIH1975,\nNCIH2009\n(No fusion)",""), limits = c("Fusion","No fusion",""))

ggsave(filename = here::here("results/organoid_crispr_plots","mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_all_nonessential.png"), plot = mean_phenotypic_kill_scores_in_CRISPR_organoid_model_of_lung_cancer_cell_lines_aggregated_all_nonessential)

```

